
<!DOCTYPE html>


<html lang="en" data-content_root="../../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>EOO results &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/2_single_context_cis/staging/eoo_analysis';</script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../1_project_overview/index.html">FunGen-xQTL atlas and companion projects</a></li>
<li class="toctree-l1"><a class="reference internal" href="../index.html">Single context cis-xQTL fine-mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../3_single_context_multigene_trans/index.html">Multi-gene and trans-xQTL fine-mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../4_multi_context/index.html">Propagation of multi-context xQTL effects</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../5_AD_xQTL_integration/index.html">Alzheimer’s disease (AD) loci integration</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5a_AD_GWAS_loci_harmonized.html">AD GWAS loci harmonization and refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5b_AD_GWAS_loci_properties.html">Refined AD GWAS loci properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5c_AD_loci_xQTL_annotated.html">AD loci “explained” by xQTL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5d_AD_loci_table_summary.html">Summary of FunGen-xQTL AD loci table</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5e_GSEA.html">Gene Set Enrichment Analysis for AD Risk Genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5f_PPI.html">PPI Networks for Selected AD Risk Genes in xQTL Loci</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5g_cV2F.html">Consensus Variant-to-Function Score for AD</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../6_AD_xQTL_genes/index.html">Insights into AD risk genes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.html">Visualization of FunGen-xQTL AD loci table</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../../7_AD_gene_prioritized/index.html">Molecular insights and gene prioritization at key AD loci</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/2_single_context_cis/staging/eoo_analysis.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../../_sources/main_text/2_single_context_cis/staging/eoo_analysis.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>EOO results</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#input">input</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-preparation">Data preparation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization">Visualization</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#prepare-data">prepare data</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#load-updated-multicontext">load updated multicontext</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#load-all-other-data">load all other data</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#combine-as-new-data">combine as new data</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plot">Plot</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#barplot">barplot</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#volcano">volcano</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#test-between-methods">test between methods</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-to-single-context">multi gene to single context</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-context-to-single-context">multi context to single context</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#summary-of-updated-multi-context-eoo-results">Summary of Updated Multi-Context EOO Results</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="eoo-results">
<h1>EOO results<a class="headerlink" href="#eoo-results" title="Link to this heading">#</a></h1>
<p>The visualization should be replicated starting from Viusalization - Plot; input data has been uploaded to s3://statfungen/ftp_fgc_xqtl/analysis_result/EOO/EOO_reshape_combined_result_withSNP_pip01_95cs_annotation_variant_level.updatedSep9th.RDS</p>
<section id="input">
<h2>input<a class="headerlink" href="#input" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># multi context to EOO input </span>

<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>

<span class="n">multi_con</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/data/ROSMAP.exported.toploci.bed.gz&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Filter rows and pull the lfsr column</span>
<span class="n">multi_con_clean</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_con</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">lfsr</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">cs_coverage_0.95</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">PIP</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0.1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multi_con_clean</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">   </span><span class="nf">pull</span><span class="p">(</span><span class="n">lfsr</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1"># split by &quot;;&quot; into matrix</span>
<span class="w">  </span><span class="nf">str_split</span><span class="p">(</span><span class="s">&quot;;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1"># convert to vector</span>
<span class="w">  </span><span class="nf">unlist</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1"># convert to numeric</span>
<span class="w">  </span><span class="nf">as.numeric</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">.</span><span class="p">[</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">.</span><span class="p">)]</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1"># show summary statistics</span>
<span class="w">  </span><span class="nf">summary</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
0.000e+00 0.000e+00 1.031e-06 8.171e-04 3.627e-04 1.000e-02 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multi_con_clean_long</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_con_clean</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1"># separate_longer_delim is tidyverse way to split long by delimiter</span>
<span class="w">  </span><span class="nf">separate_longer_delim</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">delim</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multi_con_clean_long</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_con_clean_long</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">event_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;_multicontext&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multi_con_clean_long</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">event_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;&quot;\n&quot;&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">cat</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AC_DeJager_eQTL_multicontext&quot;
&quot;DLPFC_DeJager_eQTL_multicontext&quot;
&quot;PCC_DeJager_eQTL_multicontext&quot;
&quot;Exc_mega_eQTL_multicontext&quot;
&quot;Ast_Kellis_eQTL_multicontext&quot;
&quot;Ast_mega_eQTL_multicontext&quot;
&quot;Exc_DeJager_eQTL_multicontext&quot;
&quot;Exc_Kellis_eQTL_multicontext&quot;
&quot;Inh_DeJager_eQTL_multicontext&quot;
&quot;Inh_Kellis_eQTL_multicontext&quot;
&quot;Inh_mega_eQTL_multicontext&quot;
&quot;Oli_DeJager_eQTL_multicontext&quot;
&quot;Oli_mega_eQTL_multicontext&quot;
&quot;Ast_DeJager_eQTL_multicontext&quot;
&quot;OPC_Kellis_eQTL_multicontext&quot;
&quot;DLPFC_Bennett_pQTL_multicontext&quot;
&quot;OPC_DeJager_eQTL_multicontext&quot;
&quot;Oli_Kellis_eQTL_multicontext&quot;
&quot;OPC_mega_eQTL_multicontext&quot;
&quot;monocyte_ROSMAP_eQTL_multicontext&quot;
&quot;Mic_Kellis_eQTL_multicontext&quot;
&quot;Mic_mega_eQTL_multicontext&quot;
&quot;Mic_DeJager_eQTL_multicontext
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multi_con_clean_long</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_split</span><span class="p">(</span><span class="n">event_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">walk</span><span class="p">(</span><span class="kr">function</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">id</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">event_ID</span><span class="p">)</span>
<span class="w">    </span><span class="n">file_name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/data/ROSMAP.&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">id</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.exported.toploci.tsv&quot;</span><span class="p">)</span><span class="w">  </span>
<span class="w">    </span><span class="nf">fwrite</span><span class="p">(</span><span class="n">df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="n">`#chr`</span><span class="p">,</span><span class="w"> </span><span class="n">end</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">rename</span><span class="p">(</span><span class="n">chr</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">`#chr`</span><span class="p">,</span><span class="w"> </span><span class="n">pos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">end</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">arrange</span><span class="p">(</span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="n">pos</span><span class="p">),</span><span class="w"> </span><span class="n">file_name</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
<span class="w">  </span><span class="p">})</span>
</pre></div>
</div>
</div>
</div>
<p>In the first itration, we found the multi context does not show a better enrichment than single context, we wondered if we should use a version</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Identify pure cell-specific multi-context, which in principle to keep only variants that:
•	Appear in a single context (event_ID has no ;)
•	And do not appear in any rows where event_ID includes a ; (multi-context bundles)
</pre></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Identify variant_IDs that are in multi-context bundles</span>
<span class="n">multi_context_variants</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_con_clean</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w"> </span><span class="nf">unique</span><span class="p">()</span>

<span class="c1"># Keep only rows where event_ID has no &#39;;&#39; AND variant_ID not in multi-context</span>
<span class="n">pure_cell_specific</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_con_clean</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">),</span><span class="w"> </span><span class="o">!</span><span class="n">variant_ID</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">multi_context_variants</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multi_con_clean</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>41213</li><li>14</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">pure_cell_specific</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>3126</li><li>14</li></ol>
</div></div>
</div>
<p>too few to continue EOO analysis with this pure way</p>
</section>
<section id="data-preparation">
<h2>Data preparation<a class="headerlink" href="#data-preparation" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;gz$&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">368</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">()</span>
<span class="kr">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">seq_along</span><span class="p">(</span><span class="n">eoo_res_list</span><span class="p">)){</span>
<span class="w">    </span><span class="n">eoo_res</span><span class="w">  </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">eoo_res</span><span class="p">,</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">eoo_res_list</span><span class="p">[</span><span class="n">i</span><span class="p">])</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">eoo_res_list</span><span class="p">[</span><span class="n">i</span><span class="p">])</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;[.]&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">.</span><span class="p">[,</span><span class="m">1</span><span class="p">]))</span>

<span class="p">}</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">fwrite</span><span class="p">(</span><span class="n">eoo_res</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment/ROSMAP_multicontext.all.enrichment_results_summary.tsv.gz&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 × 12</caption>
<thead>
	<tr><th scope=col>Annotation</th><th scope=col>OR</th><th scope=col>OR_SE</th><th scope=col>OR_log2</th><th scope=col>OR_SE_log2</th><th scope=col>Enrichment</th><th scope=col>Enrichment_SE</th><th scope=col>Enrichment_log2</th><th scope=col>Enrichment_SE_log2</th><th scope=col>Enrichment_Z_score</th><th scope=col>Enrichment_P_value</th><th scope=col>study</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>LHX2_TSS_H3K27ac                 </td><td>10.478451</td><td>0.4956974</td><td>3.389354</td><td>0.06813028</td><td>9.462041</td><td>0.4672775</td><td>3.242068</td><td>0.07111028</td><td>20.249298</td><td> 3.603390e-91</td><td>AC_DeJager_eQTL_multicontext</td></tr>
	<tr><td>LHX2_enhancer                    </td><td> 2.891761</td><td>0.2098354</td><td>1.531948</td><td>0.10443331</td><td>2.771898</td><td>0.1944439</td><td>1.470706</td><td>0.10097178</td><td>14.255515</td><td> 4.142337e-46</td><td>AC_DeJager_eQTL_multicontext</td></tr>
	<tr><td>LHX2_enhancer_atac               </td><td> 6.359157</td><td>0.6551940</td><td>2.668836</td><td>0.14953289</td><td>6.200055</td><td>0.6254002</td><td>2.631929</td><td>0.14642127</td><td> 9.913741</td><td> 3.628028e-23</td><td>AC_DeJager_eQTL_multicontext</td></tr>
	<tr><td>LHX2_optimal_peak.H3K27          </td><td> 5.353238</td><td>0.2439659</td><td>2.420412</td><td>0.06586453</td><td>4.689886</td><td>0.2152761</td><td>2.229480</td><td>0.06632624</td><td>21.785450</td><td>3.187966e-105</td><td>AC_DeJager_eQTL_multicontext</td></tr>
	<tr><td>LHX2_optimal_peak.H3K4me3        </td><td>10.151530</td><td>0.4857885</td><td>3.343625</td><td>0.06918673</td><td>9.274875</td><td>0.4537692</td><td>3.213245</td><td>0.07070101</td><td>20.439632</td><td> 7.429773e-93</td><td>AC_DeJager_eQTL_multicontext</td></tr>
	<tr><td>LHX2_optimal_peak_IDR_ENCODE.ATAC</td><td> 7.957453</td><td>0.4111946</td><td>2.992307</td><td>0.07456977</td><td>7.451540</td><td>0.3673269</td><td>2.897455</td><td>0.07114755</td><td>20.285857</td><td> 1.714488e-91</td><td>AC_DeJager_eQTL_multicontext</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="visualization">
<h2>Visualization<a class="headerlink" href="#visualization" title="Link to this heading">#</a></h2>
<section id="prepare-data">
<h3>prepare data<a class="headerlink" href="#prepare-data" title="Link to this heading">#</a></h3>
<section id="load-updated-multicontext">
<h4>load updated multicontext<a class="headerlink" href="#load-updated-multicontext" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">scales</span><span class="p">)</span>
<span class="n">eoo_res_multicon</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="w"> </span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment/ROSMAP_multicontext.all.enrichment_results_summary.tsv.gz&#39;</span><span class="p">)</span>

<span class="c1">#### load data ####</span>
<span class="n">meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;~/Work/gpQTL//EOO/data/realease_data_readme.txt&#39;</span><span class="p">)</span>
<span class="n">meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">context_mod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">Path</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;.exported.toploci.bed.gz&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="n">.</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_multicon</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">study</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'AC_DeJager_eQTL_multicontext'</li><li>'Ast_DeJager_eQTL_multicontext'</li><li>'Ast_Kellis_eQTL_multicontext'</li><li>'Ast_mega_eQTL_multicontext'</li><li>'DLPFC_Bennett_pQTL_multicontext'</li><li>'DLPFC_DeJager_eQTL_multicontext'</li><li>'Exc_DeJager_eQTL_multicontext'</li><li>'Exc_Kellis_eQTL_multicontext'</li><li>'Exc_mega_eQTL_multicontext'</li><li>'Inh_DeJager_eQTL_multicontext'</li><li>'Inh_Kellis_eQTL_multicontext'</li><li>'Inh_mega_eQTL_multicontext'</li><li>'Mic_DeJager_eQTL_multicontext'</li><li>'Mic_Kellis_eQTL_multicontext'</li><li>'Mic_mega_eQTL_multicontext'</li><li>'monocyte_ROSMAP_eQTL_multicontext'</li><li>'Oli_DeJager_eQTL_multicontext'</li><li>'Oli_Kellis_eQTL_multicontext'</li><li>'Oli_mega_eQTL_multicontext'</li><li>'OPC_DeJager_eQTL_multicontext'</li><li>'OPC_Kellis_eQTL_multicontext'</li><li>'OPC_mega_eQTL_multicontext'</li><li>'PCC_DeJager_eQTL_multicontext'</li></ol>
</div></div>
</div>
</section>
<section id="load-all-other-data">
<h4>load all other data<a class="headerlink" href="#load-all-other-data" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="w"> </span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/EOO_reshape_combined_result_withSNP_pip01_95cs_annotation_variant_level.RDS&#39;</span><span class="p">)</span>
<span class="c1"># eoo_res &lt;- readRDS( &#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/EOO_original_result_files_withSNP_pip01_95cs_annotation_variant_level.RDS&#39;)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">rename</span><span class="p">(</span><span class="n">Enrichment_Z_score</span><span class="o">=</span><span class="n">Z_enrich</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">P_enrich</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">method</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;multicontext&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">study</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'MSBB'</li><li>'ROSMAP'</li><li>'ROSMAP_AC'</li><li>'ROSMAP_DLPFC'</li><li>'ROSMAP_Exc'</li><li>'ROSMAP_Inh'</li><li>'ROSMAP_Mic'</li><li>'ROSMAP_OPC'</li><li>'ROSMAP_PCC'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">method</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">study</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">'ROSMAP_Ast_multicontext'</div></div>
</div>
</section>
<section id="combine-as-new-data">
<h4>combine as new data<a class="headerlink" href="#combine-as-new-data" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_all_nomulticon</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">method</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&#39;multicontext&#39;</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="nf">is.na</span><span class="p">(</span><span class="n">method</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plyr</span><span class="o">::</span><span class="nf">rbind.fill</span><span class="p">(</span><span class="n">eoo_res_all_nomulticon</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">,</span><span class="w"> </span><span class="n">eoo_res_multicon</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;multicontext&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">Annotation</span><span class="o">!=</span><span class="s">&#39;UTR_5_UCSC.to_hg38&#39;</span><span class="p">))</span><span class="w"> </span><span class="c1"># a duplicate annotation</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 17</caption>
<thead>
	<tr><th></th><th scope=col>Annotation</th><th scope=col>OR</th><th scope=col>OR_SE</th><th scope=col>OR_log2</th><th scope=col>OR_SE_log2</th><th scope=col>Enrichment</th><th scope=col>Enrichment_SE</th><th scope=col>Enrichment_log2</th><th scope=col>Enrichment_SE_log2</th><th scope=col>study</th><th scope=col>method</th><th scope=col>annotation</th><th scope=col>annotation_sets</th><th scope=col>Enrichment_Z_score</th><th scope=col>Enrichment_P_value</th><th scope=col>Z_or</th><th scope=col>P_or</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>1</th><td>LHX2_TSS_H3K27ac                 </td><td>12.723741</td><td>0.7243419</td><td>3.669451</td><td>0.08187278</td><td>11.352247</td><td>0.6569629</td><td>3.504791</td><td>0.08324203</td><td>AC_DeJager_eQTL</td><td>univariate</td><td>AI_MPRA_starrseq_baseline_1000G_EUR_hg38_annotations</td><td>AI_MPRA_starrseq</td><td>15.757733</td><td>0.000000e+00</td><td>44.81894</td><td>0</td></tr>
	<tr><th scope=row>2</th><td>LHX2_enhancer                    </td><td> 2.973535</td><td>0.2473014</td><td>1.572179</td><td>0.12043096</td><td> 2.850839</td><td>0.2272655</td><td>1.511167</td><td>0.11543278</td><td>AC_DeJager_eQTL</td><td>univariate</td><td>AI_MPRA_starrseq_baseline_1000G_EUR_hg38_annotations</td><td>AI_MPRA_starrseq</td><td> 8.143953</td><td>4.440892e-16</td><td>13.05461</td><td>0</td></tr>
	<tr><th scope=row>3</th><td>LHX2_enhancer_atac               </td><td> 6.056163</td><td>0.7897873</td><td>2.598404</td><td>0.18915166</td><td> 5.945633</td><td>0.7606750</td><td>2.571264</td><td>0.18561522</td><td>AC_DeJager_eQTL</td><td>univariate</td><td>AI_MPRA_starrseq_baseline_1000G_EUR_hg38_annotations</td><td>AI_MPRA_starrseq</td><td> 6.501637</td><td>7.945045e-11</td><td>13.73715</td><td>0</td></tr>
	<tr><th scope=row>4</th><td>LHX2_optimal_peak.H3K27          </td><td> 6.227424</td><td>0.3309881</td><td>2.638635</td><td>0.07691982</td><td> 5.346442</td><td>0.2765385</td><td>2.418487</td><td>0.07482234</td><td>AC_DeJager_eQTL</td><td>univariate</td><td>AI_MPRA_starrseq_baseline_1000G_EUR_hg38_annotations</td><td>AI_MPRA_starrseq</td><td>15.717315</td><td>0.000000e+00</td><td>34.30371</td><td>0</td></tr>
	<tr><th scope=row>5</th><td>LHX2_optimal_peak.H3K4me3        </td><td>13.591981</td><td>0.7181150</td><td>3.764684</td><td>0.07614929</td><td>12.169811</td><td>0.6565377</td><td>3.605135</td><td>0.07771441</td><td>AC_DeJager_eQTL</td><td>univariate</td><td>AI_MPRA_starrseq_baseline_1000G_EUR_hg38_annotations</td><td>AI_MPRA_starrseq</td><td>17.013206</td><td>0.000000e+00</td><td>49.43820</td><td>0</td></tr>
	<tr><th scope=row>6</th><td>LHX2_optimal_peak_IDR_ENCODE.ATAC</td><td>10.153046</td><td>0.5578225</td><td>3.343841</td><td>0.07882028</td><td> 9.403917</td><td>0.4846550</td><td>3.233171</td><td>0.07397257</td><td>AC_DeJager_eQTL</td><td>univariate</td><td>AI_MPRA_starrseq_baseline_1000G_EUR_hg38_annotations</td><td>AI_MPRA_starrseq</td><td>17.340000</td><td>0.000000e+00</td><td>42.42361</td><td>0</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">saveRDS</span><span class="p">(</span><span class="n">eoo_res_all</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/EOO_reshape_combined_result_withSNP_pip01_95cs_annotation_variant_level.updatedSep9th.RDS&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># # reorganize track df </span>
<span class="c1"># old_track_df &lt;- fread(&quot;/Work/Fungen_xQTL/2025/Sep/eoo/EOO/data/categorized_annotation_tracks.csv&quot;)</span>
<span class="c1"># encode_track_df &lt;- fread(&quot;/Work/Fungen_xQTL/2025/Sep/eoo/EOO/data/encode_categories_Ru.csv&quot;)</span>

<span class="c1"># encode_track_df &lt;- encode_track_df %&gt;% select(V1, V2)</span>
<span class="c1"># encode_track_df &lt;- encode_track_df[-1,] %&gt;% mutate(Source = &#39;ENCODE&#39;)</span>
<span class="c1"># colnames(encode_track_df) &lt;- c(&#39;Track&#39;, &#39;Category&#39;, &#39;Source&#39;)</span>
<span class="c1"># track_df &lt;- plyr::rbind.fill(old_track_df, encode_track_df)</span>
<span class="c1"># fwrite(track_df, &quot;/Work/Fungen_xQTL/2025/Sep/eoo/EOO/data/categorized_annotation_tracks.csv&quot;)</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="plot">
<h3>Plot<a class="headerlink" href="#plot" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&#39;EOO_reshape_combined_result_withSNP_pip01_95cs_annotation_variant_level.updatedSep9th.RDS&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">enrichment_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">eoo_res_all</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;.to_hg38&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="s">&#39;hg19&#39;</span><span class="p">))</span>

<span class="n">enrichment_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span>
<span class="w">    </span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ENCFF038KTR|ENCFF015DGX|ENCFF370ANN&#39;</span><span class="p">),</span>
<span class="w">    </span><span class="nf">str_replace</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;\\.bed$&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.bed.gz&#39;</span><span class="p">),</span>
<span class="w">    </span><span class="n">Annotation</span><span class="w">  </span><span class="c1"># keep original if condition not met</span>
<span class="w">  </span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">distinct</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">track_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;categorized_annotation_tracks.csv&quot;</span><span class="p">)</span>

<span class="n">enrichment_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">enrichment_data</span><span class="p">,</span><span class="w"> </span><span class="n">track_df</span><span class="p">,</span><span class="w"> </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Annotation&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Track&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">all.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">Source</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">distinct</span>

<span class="c1"># enrichment_data &lt;- enrichment_data %&gt;%mutate(Annotation = ifelse(str_detect(Annotation, &#39;ENCFF&#39;),paste0(Category, &quot;_&quot;, Annotation),Annotation))</span>
<span class="n">enrichment_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">Category</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;|&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="p">))</span>

<span class="c1"># Reorder Annotation based on the order of Category</span>
<span class="n">enrichment_data</span><span class="o">$</span><span class="n">Annotation</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">enrichment_data</span><span class="o">$</span><span class="n">Annotation</span><span class="p">,</span>
<span class="w">                                     </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">enrichment_data</span><span class="o">$</span><span class="n">Annotation</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="n">enrichment_data</span><span class="o">$</span><span class="n">Category</span><span class="p">)]))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">enrichment_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 24</caption>
<thead>
	<tr><th></th><th scope=col>Annotation</th><th scope=col>OR</th><th scope=col>OR_SE</th><th scope=col>OR_log2</th><th scope=col>OR_SE_log2</th><th scope=col>Enrichment</th><th scope=col>Enrichment_SE</th><th scope=col>Enrichment_log2</th><th scope=col>Enrichment_SE_log2</th><th scope=col>study</th><th scope=col>⋯</th><th scope=col>Enrichment_P_value</th><th scope=col>Z_or</th><th scope=col>P_or</th><th scope=col>Assay</th><th scope=col>Data_Category</th><th scope=col>Cell_Type</th><th scope=col>Category</th><th scope=col>Annotation_Category</th><th scope=col>Priority</th><th scope=col>Notes</th></tr>
	<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>1</th><td>Transcription_Factor_Binding|1_GATA2_1-Interval-Track</td><td>1.809502</td><td>0.11650544</td><td>0.8555927</td><td>0.09286834</td><td>1.783705</td><td>0.11105414</td><td>0.8347443</td><td>0.08981472</td><td>Oli_DeJager_eQTL_multicontext    </td><td>⋯</td><td> 4.742789e-58</td><td>NA</td><td>NA</td><td></td><td></td><td></td><td>Transcription_Factor_Binding</td><td>Uncategorized</td><td>Exploratory</td><td>Not directly mapped; exploratory</td></tr>
	<tr><th scope=row>2</th><td>Transcription_Factor_Binding|1_GATA2_1-Interval-Track</td><td>1.737711</td><td>0.10442133</td><td>0.7971878</td><td>0.08647314</td><td>1.714913</td><td>0.09971817</td><td>0.7780193</td><td>0.08368707</td><td>Oli_Kellis_eQTL_multicontext     </td><td>⋯</td><td> 2.767894e-66</td><td>NA</td><td>NA</td><td></td><td></td><td></td><td>Transcription_Factor_Binding</td><td>Uncategorized</td><td>Exploratory</td><td>Not directly mapped; exploratory</td></tr>
	<tr><th scope=row>3</th><td>Transcription_Factor_Binding|1_GATA2_1-Interval-Track</td><td>1.816044</td><td>0.11174689</td><td>0.8607993</td><td>0.08874149</td><td>1.790219</td><td>0.10689954</td><td>0.8400134</td><td>0.08612717</td><td>OPC_Kellis_eQTL_multicontext     </td><td>⋯</td><td> 5.981670e-63</td><td>NA</td><td>NA</td><td></td><td></td><td></td><td>Transcription_Factor_Binding</td><td>Uncategorized</td><td>Exploratory</td><td>Not directly mapped; exploratory</td></tr>
	<tr><th scope=row>4</th><td>Transcription_Factor_Binding|1_GATA2_1-Interval-Track</td><td>1.737365</td><td>0.22948146</td><td>0.7969008</td><td>0.19164379</td><td>1.715772</td><td>0.21975690</td><td>0.7782897</td><td>0.18587504</td><td>monocyte_ROSMAP_eQTL_multicontext</td><td>⋯</td><td> 5.829208e-15</td><td>NA</td><td>NA</td><td></td><td></td><td></td><td>Transcription_Factor_Binding</td><td>Uncategorized</td><td>Exploratory</td><td>Not directly mapped; exploratory</td></tr>
	<tr><th scope=row>5</th><td>Transcription_Factor_Binding|1_GATA2_1-Interval-Track</td><td>1.769361</td><td>0.08809096</td><td>0.8232284</td><td>0.07188886</td><td>1.744402</td><td>0.08413339</td><td>0.8026526</td><td>0.06964766</td><td>Exc_DeJager_eQTL_multicontext    </td><td>⋯</td><td> 1.718045e-95</td><td>NA</td><td>NA</td><td></td><td></td><td></td><td>Transcription_Factor_Binding</td><td>Uncategorized</td><td>Exploratory</td><td>Not directly mapped; exploratory</td></tr>
	<tr><th scope=row>6</th><td>Transcription_Factor_Binding|1_GATA2_1-Interval-Track</td><td>1.604405</td><td>0.06880918</td><td>0.6820384</td><td>0.06174599</td><td>1.585853</td><td>0.06590343</td><td>0.6651996</td><td>0.05983695</td><td>PCC_DeJager_eQTL_multicontext    </td><td>⋯</td><td>6.061492e-128</td><td>NA</td><td>NA</td><td></td><td></td><td></td><td>Transcription_Factor_Binding</td><td>Uncategorized</td><td>Exploratory</td><td>Not directly mapped; exploratory</td></tr>
</tbody>
</table>
</div></div>
</div>
<section id="barplot">
<h4>barplot<a class="headerlink" href="#barplot" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Get original Set2 colors (8)</span>
<span class="n">set2_colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Set3&quot;</span><span class="p">)</span>

<span class="c1"># Generate extra colors in a similar style</span>
<span class="n">n_cat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">enrichment_data</span><span class="o">$</span><span class="n">Category</span><span class="p">))</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">n_cat</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">8</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">extra_colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">hue_pal</span><span class="p">(</span><span class="n">h</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="m">300</span><span class="p">),</span><span class="w"> </span><span class="n">c</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">70</span><span class="p">,</span><span class="w"> </span><span class="n">l</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">70</span><span class="p">)(</span><span class="n">n_cat</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">8</span><span class="p">)</span>
<span class="w">  </span><span class="n">all_colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">set2_colors</span><span class="p">,</span><span class="w"> </span><span class="n">extra_colors</span><span class="p">)</span>
<span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">all_colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">set2_colors</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="n">n_cat</span><span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="o">=</span><span class="m">10</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">##### seperate figures #####</span>
<span class="kr">for</span><span class="p">(</span><span class="n">context</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">enrichment_data</span><span class="o">$</span><span class="n">study</span><span class="p">)){</span>
<span class="w">  </span><span class="nf">print</span><span class="p">(</span><span class="n">context</span><span class="p">)</span>
<span class="w">  </span><span class="n">dir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">context_mod</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">Method</span><span class="p">)</span>
<span class="w">  </span><span class="kr">if</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">dir</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="n">dir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Other&#39;</span>
<span class="w">  </span><span class="n">dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">filter</span><span class="p">(</span>
<span class="w">        </span><span class="n">study</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">,</span>
<span class="w">        </span><span class="n">Enrichment_P_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span>
<span class="w">      </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span><span class="c1"># Only significant</span>
<span class="w">      </span><span class="nf">group_by</span><span class="p">(</span><span class="n">Category</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">Enrichment_log2</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">        </span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">reorder</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="p">),</span><span class="w">  </span><span class="c1"># Fix order ONCE</span>
<span class="w">        </span><span class="n">ymin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">Enrichment_SE_log2</span><span class="p">,</span>
<span class="w">        </span><span class="n">ymax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">Enrichment_SE_log2</span>
<span class="w">      </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">distinct</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">.keep_all</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">  </span><span class="c1"># Remove duplicates so bars don&#39;t stack</span>
<span class="w">    </span>
<span class="w">    </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Category</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_col</span><span class="p">(</span><span class="n">stat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;identity&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_errorbar</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">ymin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ymin</span><span class="p">,</span><span class="w"> </span><span class="n">ymax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ymax</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.3</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">theme_bw</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">labs</span><span class="p">(</span>
<span class="w">        </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;Enrichment across Annotations: &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">context</span><span class="p">),</span>
<span class="w">        </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Annotation&quot;</span><span class="p">,</span>
<span class="w">        </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log₂(Enrichment Score)&quot;</span><span class="p">,</span>
<span class="w">        </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Annotation Set&quot;</span>
<span class="w">      </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">all_colors</span><span class="p">)</span>
<span class="w">  </span><span class="n">output</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/&#39;</span><span class="p">,</span><span class="n">dir</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;_enrichment_plot.pdf&#39;</span><span class="p">)</span>
<span class="w">  </span><span class="nf">dir.create</span><span class="p">(</span><span class="nf">dirname</span><span class="p">(</span><span class="n">output</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="w">  </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">output</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">50</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Oli_DeJager_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Oli_Kellis_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;OPC_Kellis_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;monocyte_ROSMAP_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Exc_DeJager_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;PCC_DeJager_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Ast_DeJager_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Oli_mega_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Mic_mega_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;DLPFC_DeJager_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Exc_Kellis_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Inh_Kellis_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Inh_mega_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Mic_Kellis_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;OPC_mega_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Mic_DeJager_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;AC_DeJager_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;OPC_DeJager_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Ast_Kellis_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;DLPFC_Bennett_pQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Inh_DeJager_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Exc_mega_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Ast_mega_eQTL_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Ast_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;OPC_Kellis_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_AC_sQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Metabrain_Cortex&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_AC&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 144 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_Exc&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 164 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MiGA_eQTL_THA&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 131 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_pQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 78 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_OPC_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 8 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MiGA_GTS_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MiGA_eQTL_SVZ&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 125 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;DLPFC_Klein_gpQTL_adjusted&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 21 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_eQTL_BM44&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 134 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Ast_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 3 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Kelli_Inh&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 147 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Ast_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MiGA_GFM_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_eQTL_BM22&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 112 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Mic_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_BM36_pQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Ast_10_Kellis_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 23 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DLPFC_Bennett_pQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Metabrain_Spinalcord&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 4 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Mic_xqtl_only_colocboost&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_BM_22_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 16 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;DLPFC_Klein_gpQTL_unadjusted&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 13 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;monocyte_ROSMAP_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;OPC_xqtl_only_colocboost&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MiGA_SVZ_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Mic_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 64 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_Oli&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 164 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Knight_pQTL_brain&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 20 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_metabolome&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 9 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_mega_Oli&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 163 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Inh_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_DLPFC&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 49 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Oli_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Oli_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Mic_12_Kellis_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 97 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DLPFC_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Knight_eQTL_brain&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_Mic&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 171 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Exc_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Exc_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_mega_Mic&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 175 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_OPC_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 61 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Mic_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 61 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_eQTL_BM36&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 144 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;DLPFC_xqtl_only_colocboost&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;DLPFC_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_pQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 58 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Mic_Kellis_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Ast_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Inh_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MiGA_THA_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_all_cs_snp_mQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_PCC&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 99 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_Ast&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 159 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;BM_22_MSBB_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_mQTL.top_pc_1.cs_snp&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_PCC_sQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;STARNET_eQTL_Mac&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Inh_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;PCC_xqtl_only_colocboost&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Kelli_Ast&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 154 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_mega_OPC&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 169 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Oli_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_Inh&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 157 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Inh_Kellis_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_all_cs_snp_haQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;BM_36_MSBB_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;BM_10_MSBB_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Knight_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 124 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Ast_Kellis_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Kelli_OPC&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 141 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Metabrain_Basalganglia&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_AC_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_mega_Ast&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 170 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;DLPFC_Bennett_pQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Exc_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_BM_44_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 18 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Oli_Kellis_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Kelli_Mic&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 174 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Exc_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Exc_Kellis_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Knight_pQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 172 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;AC_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;AC_xqtl_only_colocboost&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Metabrain_Hippocampus&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;OPC_mega_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;OPC_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Ast_xqtl_only_colocboost&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MiGA_eQTL_GFM&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 149 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_mega_Inh&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 161 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;BM_44_MSBB_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Oli_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_mega_Exc&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 160 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_monocyte&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 114 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Mic_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Metabrain_Cerebellum&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_PCC_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MiGA_eQTL_GTS&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 131 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Exc_xqtl_only_colocboost&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_monocyte_ROSMAP_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_BM_10_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 9 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_eQTL_BM10&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 128 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;PCC_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;MSBB_BM_36_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 47 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Kelli_Oli&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 169 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Inh_DeJager_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/multi_gene_finemapping&#39; already exists”
Warning message:
“Removed 3 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_haQTL.top_pc_1.cs_snp&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message:
“Removed 17 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Kelli_Exc&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 157 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_Ast_multicontext&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;STARNET_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 134 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DeJager_OPC&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/trans_finemapping&#39; already exists”
Warning message:
“Removed 165 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Mic_13_Kellis_eQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message:
“Removed 35 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;ROSMAP_DLPFC_sQTL&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/single_context_finemapping&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Inh_xqtl_only_colocboost&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;xqtl_only_colocboost&quot;
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in geom_col(stat = &quot;identity&quot;):
“Ignoring unknown parameters: `stat`”
Warning message in dir.create(dirname(output), recursive = T):
“&#39;/home/rf2872/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/Other&#39; already exists”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">##### plot top all by category #####</span>
<span class="n">top_num</span><span class="o">=</span><span class="m">2</span>
<span class="kr">for</span><span class="p">(</span><span class="n">top_num</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">)){</span>
<span class="w">  </span><span class="n">enrichment_data_top_category</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="n">Enrichment_P_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span><span class="w"> </span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ENCFF&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span><span class="c1"># optional: focus on significant enrichments </span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span>
<span class="w">                               </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">Annotation</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="n">Category</span><span class="p">)])))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">group_by</span><span class="p">(</span><span class="n">Category</span><span class="p">,</span><span class="w"> </span><span class="n">study</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">Enrichment_log2</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">top_num</span><span class="p">)</span><span class="w">   </span><span class="c1"># optional: focus on significant enrichments</span>
<span class="w">  </span>
<span class="w">  </span>
<span class="w">  </span><span class="n">enrichment_data_top_category</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">enrichment_data_top_category</span><span class="p">,</span><span class="w"> </span><span class="n">meta</span><span class="p">,</span><span class="w"> </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;study&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;context_mod&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">all.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">`Data Type`</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;Other&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="p">),</span>
<span class="w">           </span><span class="n">Method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Method</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;Other&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">Method</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">Method</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;single_context_finemapping&#39;</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Modality</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Inh|Exc|OPC|Oli|Mic|Ast&#39;</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;snuc_eQTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">),</span>
<span class="w">           </span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">Method</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;multi_gene_finemapping&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;multi_gene&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">),</span>
<span class="w">           </span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">method</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;multicontext&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;multi_context&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">),</span>
<span class="w">           </span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;haQTL|mQTL&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Epi&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">))</span>
<span class="w">  </span><span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_top_category</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">Method</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&#39;trans_finemapping&#39;</span><span class="p">)</span>
<span class="w">  </span><span class="c1"># plot_data &lt;- enrichment_data_top_category</span>
<span class="w">  </span>
<span class="w">  </span>
<span class="w">  </span><span class="n">dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;condense_top&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">top_num</span><span class="p">)</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">data_type</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">na.omit</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">`Data Type`</span><span class="p">)),</span><span class="w"> </span><span class="s">&#39;all&#39;</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="c1"># for (data_type in c(na.omit(unique(plot_data$`Data Type`)), &#39;all&#39;)) {</span>
<span class="w">  </span><span class="c1"># for (data_type in c(&#39;all&#39;)) {</span>
<span class="w">    </span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">data_type</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;all&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data</span>
<span class="w">      </span><span class="c1"># width &lt;- height &lt;- 80</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">data_type</span><span class="p">)</span>
<span class="w">      </span><span class="c1"># width &lt;- height &lt;- 40</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annotation</span><span class="p">,</span>
<span class="w">                 </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="p">,</span>
<span class="w">                 </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Category</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_col</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_errorbar</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">ymin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">Enrichment_SE_log2</span><span class="p">,</span>
<span class="w">                        </span><span class="n">ymax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">Enrichment_SE_log2</span><span class="p">),</span>
<span class="w">                    </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.3</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;free_y&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">theme_bw</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Enrichment across Annotations&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Annotation&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log₂(Enrichment Score)&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Annotation Set&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">all_colors</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">output</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">dir</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">data_type</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;_top&#39;</span><span class="p">,</span><span class="n">top_num</span><span class="p">,</span><span class="s">&#39;_enrichment_plot_combined.pdf&#39;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">dir.create</span><span class="p">(</span><span class="nf">dirname</span><span class="p">(</span><span class="n">output</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">data_type</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;all&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">width</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">100</span>
<span class="w">      </span><span class="n">height</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">40</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">top_num</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="c1"># Dynamically calculate size based on the number of facets</span>
<span class="w">      </span><span class="n">num_facets</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">`Data Type`</span><span class="p">))</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">study</span><span class="p">))</span>
<span class="w">      </span><span class="kr">if</span><span class="p">(</span><span class="n">num_facets</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">top_num</span><span class="w"> </span><span class="kr">else</span><span class="w">  </span><span class="n">height</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">top_num</span><span class="o">/</span><span class="m">2</span><span class="w"> </span><span class="c1"># adjust multiplier as needed for better sizing</span>
<span class="w">      </span><span class="kr">if</span><span class="p">(</span><span class="n">num_facets</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">2.5</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">num_facets</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">6</span><span class="w">  </span><span class="c1"># adjust multiplier as needed for better sizing</span>
<span class="w">    </span><span class="p">}</span><span class="w">  </span>
<span class="w">    </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">output</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">height</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">width</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 6 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 4 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 13 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 25 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 14 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 8 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 34 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 2 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 60 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">##### plot top all by category for cell type marker #####</span>
<span class="n">top_num</span><span class="o">=</span><span class="m">2</span>
<span class="kr">for</span><span class="p">(</span><span class="n">top_num</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">5</span><span class="p">){</span>
<span class="w">  </span><span class="n">enrichment_data_top_category</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="n">Enrichment_P_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span><span class="w"> </span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;CellType&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span><span class="c1"># optional: focus on significant enrichments </span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span>
<span class="w">                               </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">Annotation</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="n">Category</span><span class="p">)])))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">group_by</span><span class="p">(</span><span class="n">Category</span><span class="p">,</span><span class="w"> </span><span class="n">study</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">Enrichment_log2</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">top_num</span><span class="p">)</span><span class="w">   </span><span class="c1"># optional: focus on significant enrichments</span>
<span class="w">  </span>
<span class="w">  </span>
<span class="w">  </span><span class="n">enrichment_data_top_category</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">enrichment_data_top_category</span><span class="p">,</span><span class="w"> </span><span class="n">meta</span><span class="p">,</span><span class="w"> </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;study&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;context_mod&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">all.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;LR&#39;</span><span class="p">,</span>
<span class="w">           </span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;AC|PCC|DLPFC|MSBB|Knight&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&#39;eQTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">),</span>
<span class="w">           </span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Inh|Exc|Oli|OPC|Mic|Ast|monocyte&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&#39;snuc_eQTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">),</span>
<span class="w">           </span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;pQTL&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&#39;pQTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">)</span>
<span class="w">           </span><span class="p">)</span>
<span class="w">  </span><span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_top_category</span>
<span class="w">  </span>
<span class="w">  </span>
<span class="w">  </span><span class="n">dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;condense_top&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">top_num</span><span class="p">)</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">data_type</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">na.omit</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">`Data Type`</span><span class="p">))))</span><span class="w"> </span><span class="p">{</span>

<span class="w">  </span><span class="c1"># for (data_type in c(na.omit(unique(plot_data$`Data Type`)), &#39;all&#39;)) {</span>

<span class="w">    </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">data_type</span><span class="p">)</span>

<span class="w">    </span>
<span class="w">    </span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annotation</span><span class="p">,</span>
<span class="w">                 </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="p">,</span>
<span class="w">                 </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Category</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_col</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_errorbar</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">ymin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">Enrichment_SE_log2</span><span class="p">,</span>
<span class="w">                        </span><span class="n">ymax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">Enrichment_SE_log2</span><span class="p">),</span>
<span class="w">                    </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.3</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;free_y&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">theme_bw</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Enrichment across Annotations&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Annotation&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log₂(Enrichment Score)&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Annotation Set&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">all_colors</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">output</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">dir</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">data_type</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;_top&#39;</span><span class="p">,</span><span class="n">top_num</span><span class="p">,</span><span class="s">&#39;_withENCFF_enrichment_plot_combined.pdf&#39;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">dir.create</span><span class="p">(</span><span class="nf">dirname</span><span class="p">(</span><span class="n">output</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">      </span><span class="c1"># Dynamically calculate size based on the number of facets</span>
<span class="w">      </span><span class="n">num_facets</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">`Data Type`</span><span class="p">))</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">study</span><span class="p">))</span>
<span class="w">      </span><span class="kr">if</span><span class="p">(</span><span class="n">num_facets</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">top_num</span><span class="o">/</span><span class="m">4</span><span class="w"> </span><span class="kr">else</span><span class="w">  </span><span class="n">height</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">top_num</span><span class="o">/</span><span class="m">8</span><span class="w"> </span><span class="c1"># adjust multiplier as needed for better sizing</span>
<span class="w">      </span><span class="kr">if</span><span class="p">(</span><span class="n">num_facets</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">2.5</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">num_facets</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">num_facets</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">6</span><span class="w">  </span><span class="c1"># adjust multiplier as needed for better sizing</span>
<span class="w">    </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">output</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">height</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">width</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“Removed 219 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 760 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 49 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
Warning message:
“Removed 149 rows containing missing values or values outside the scale range
(`geom_col()`).”
Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
“conversion failure on &#39;log₂(Enrichment Score)&#39; in &#39;mbcsToSbcs&#39;: for ₂ (U+2082)”
</pre></div>
</div>
</div>
</div>
</section>
<section id="volcano">
<h4>volcano<a class="headerlink" href="#volcano" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">##### valcano without Inf logP #####</span>
<span class="n">enrichment_data_meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">enrichment_data</span><span class="p">,</span><span class="w"> </span><span class="n">meta</span><span class="p">,</span><span class="w"> </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;study&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;context_mod&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">all.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">`Data Type`</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;Other&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="p">),</span>
<span class="w">         </span><span class="n">Method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Method</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;Other&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">Method</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">Method</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;single_context_finemapping&#39;</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Modality</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Inh|Exc|OPC|Oli|Mic|Ast&#39;</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;snuc_eQTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">),</span>
<span class="w">         </span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">Method</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;multi_gene_finemapping&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;multi_gene&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">),</span>
<span class="w">         </span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">method</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;multicontext&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;multi_context&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">),</span>
<span class="w">         </span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;haQTL|mQTL&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Epi&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">`Data Type`</span><span class="w"> </span><span class="p">))</span>
<span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">Method</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&#39;trans_finemapping&#39;</span><span class="p">)</span>



<span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_meta</span><span class="w"> </span>


<span class="n">dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;volcano&#39;</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">data_type</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">na.omit</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">`Data Type`</span><span class="p">)),</span><span class="w"> </span><span class="s">&#39;all&#39;</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">print</span><span class="p">(</span><span class="n">data_type</span><span class="p">)</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">data_type</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;all&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data</span>
<span class="w">    </span><span class="c1"># width &lt;- height &lt;- 80</span>
<span class="w">  </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">data_type</span><span class="p">)</span>
<span class="w">    </span><span class="c1"># width &lt;- height &lt;- 40</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Category</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Enrichment_P_value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">logP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">Enrichment_P_value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="c1"># Add a column to identify points to label (top 5 by logP)</span>
<span class="w">    </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logP</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logP</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logP</span><span class="p">),</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">facet_grid</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">study</span><span class="w"> </span><span class="o">~</span><span class="n">Category</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;fixed&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">scale_color_gradientn</span><span class="p">(</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">),</span>
<span class="w">                          </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;-log10(P)&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">scale_size_continuous</span><span class="p">(</span><span class="n">range</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">6</span><span class="p">),</span><span class="w"> </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;-log10(P)&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme_bw</span><span class="p">(</span><span class="n">base_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">14</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme</span><span class="p">(</span>
<span class="w">      </span><span class="n">panel.grid.minor</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_blank</span><span class="p">(),</span>
<span class="w">      </span><span class="n">panel.grid.major</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_line</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey90&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">panel.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">),</span>
<span class="w">      </span><span class="n">plot.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">),</span>
<span class="w">      </span><span class="n">strip.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey95&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey70&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">),</span>
<span class="w">      </span><span class="n">axis.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">axis.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;right&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">legend.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">)</span>
<span class="w">    </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">labs</span><span class="p">(</span>
<span class="w">      </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Volcano Plot: Enrichment Analysis&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">subtitle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log2(Enrichment) vs -log10(P-value)&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log2(Enrichment)&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;-log10(P-value)&quot;</span>
<span class="w">    </span><span class="p">)</span>
<span class="w">  </span><span class="n">output</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">dir</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">data_type</span><span class="p">,</span><span class="s">&#39;_enrichment_volcano_plot_combined.pdf&#39;</span><span class="p">)</span>
<span class="w">  </span><span class="nf">dir.create</span><span class="p">(</span><span class="nf">dirname</span><span class="p">(</span><span class="n">output</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># if (data_type == &#39;all&#39;) {</span>
<span class="w">  </span><span class="n">width</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">45</span>
<span class="w">  </span><span class="n">height</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2.5</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">study</span><span class="p">))</span>
<span class="w">  </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">output</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">height</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">width</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="p">}</span>

<span class="c1">##### valcano with Inf logP #####</span>


<span class="n">dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;volcano_Inf&#39;</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">data_type</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">na.omit</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">`Data Type`</span><span class="p">)),</span><span class="w"> </span><span class="s">&#39;all&#39;</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">print</span><span class="p">(</span><span class="n">data_type</span><span class="p">)</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">data_type</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;all&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data</span>
<span class="w">    </span><span class="c1"># width &lt;- height &lt;- 80</span>
<span class="w">  </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">data_type</span><span class="p">)</span>
<span class="w">    </span><span class="c1"># width &lt;- height &lt;- 40</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span><span class="c1"># Filter and compute logP</span>
<span class="w">  </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Category</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Enrichment_P_value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">logP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">Enrichment_P_value</span><span class="p">))</span>
<span class="w">  </span><span class="c1"># Calculate 98th percentile of finite logP</span>
<span class="w">  </span><span class="n">logP_top</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">quantile</span><span class="p">(</span>
<span class="w">    </span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">logP</span><span class="p">[</span><span class="nf">is.finite</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">logP</span><span class="p">)],</span><span class="w"> </span>
<span class="w">    </span><span class="m">0.99</span><span class="p">,</span><span class="w"> </span>
<span class="w">    </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span>
<span class="w">  </span><span class="p">)</span>
<span class="w">  </span><span class="n">enrich_top</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">quantile</span><span class="p">(</span>
<span class="w">    </span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">Enrichment_log2</span><span class="p">[</span><span class="nf">is.finite</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">Enrichment_log2</span><span class="p">)],</span><span class="w"> </span>
<span class="w">    </span><span class="m">0.95</span><span class="p">,</span><span class="w"> </span>
<span class="w">    </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span>
<span class="w">  </span><span class="p">)</span>
<span class="w">  </span><span class="c1"># Compute the cap and jitter for Inf values</span>
<span class="w">  </span><span class="n">finite_logP</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">logP</span><span class="p">[</span><span class="nf">is.finite</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">logP</span><span class="p">)]</span>
<span class="w">  </span><span class="n">cap_base</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">finite_logP</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">10</span>
<span class="w">  </span><span class="n">n_inf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="nf">is.infinite</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">logP</span><span class="p">),</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">  </span><span class="n">jitter_values</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">runif</span><span class="p">(</span><span class="n">n_inf</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">5</span><span class="p">)</span>
<span class="w">  </span><span class="c1"># Counter for jitter values</span>
<span class="w">  </span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">jitter_index</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NA</span>
<span class="w">  </span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">jitter_index</span><span class="p">[</span><span class="nf">is.infinite</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">logP</span><span class="p">)]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">seq_len</span><span class="p">(</span><span class="n">n_inf</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Apply cap and jitter</span>
<span class="w">  </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">logP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">      </span><span class="nf">is.infinite</span><span class="p">(</span><span class="n">logP</span><span class="p">)</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">cap_base</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">jitter_values</span><span class="p">[</span><span class="n">jitter_index</span><span class="p">],</span>
<span class="w">      </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">logP</span>
<span class="w">    </span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">jitter_index</span><span class="p">)</span><span class="w">  </span><span class="c1"># Clean up temporary column</span>
<span class="w">  </span>
<span class="w">  </span>
<span class="w">  </span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data_tmp</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">highlight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logP</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="n">logP_top</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">enrich_top</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_log2</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logP</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logP</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logP</span><span class="p">),</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="c1"># Add red circle outlines around highlighted points</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="nf">filter</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">highlight</span><span class="p">),</span><span class="w"> </span>
<span class="w">               </span><span class="n">shape</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">21</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;#CD2626&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">               </span><span class="nf">aes</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logP</span><span class="p">),</span><span class="w"> </span><span class="n">stroke</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">facet_grid</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">study</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Category</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;fixed&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">               </span><span class="n">labeller</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">labeller</span><span class="p">(</span><span class="n">Category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">label_wrap_gen</span><span class="p">(</span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">)))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">scale_color_gradientn</span><span class="p">(</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;#FF6A6A&quot;</span><span class="p">),</span>
<span class="w">                          </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;-log10(P)&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">scale_size_continuous</span><span class="p">(</span><span class="n">range</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">6</span><span class="p">),</span><span class="w"> </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;-log10(P)&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">coord_cartesian</span><span class="p">(</span><span class="n">ylim</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">logP</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">1.1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme_bw</span><span class="p">(</span><span class="n">base_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">14</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme</span><span class="p">(</span>
<span class="w">      </span><span class="n">panel.grid.minor</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_blank</span><span class="p">(),</span>
<span class="w">      </span><span class="n">panel.grid.major</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_line</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey90&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">panel.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">),</span>
<span class="w">      </span><span class="n">plot.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">),</span>
<span class="w">      </span><span class="n">strip.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey95&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey70&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">),</span>
<span class="w">      </span><span class="n">axis.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">axis.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;right&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">legend.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">)</span>
<span class="w">    </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">labs</span><span class="p">(</span>
<span class="w">      </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;Volcano Plot: Enrichment Analysis (&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">data_type</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;)&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">subtitle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log2(Enrichment) vs -log10(P-value)&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log2(Enrichment)&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;-log10(P-value)&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">caption</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Red circles: Top 1% -log10(P) &amp; Top 5% log2(Enrichment) \nP=0 capped at max finite -log10(P) + 10 with jitter&quot;</span>
<span class="w">    </span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="n">output</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/enrichment_figures_category/&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">dir</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">data_type</span><span class="p">,</span><span class="s">&#39;_enrichment_volcano_plot_combined.pdf&#39;</span><span class="p">)</span>
<span class="w">  </span><span class="nf">dir.create</span><span class="p">(</span><span class="nf">dirname</span><span class="p">(</span><span class="n">output</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="n">width</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">45</span>
<span class="w">  </span><span class="n">height</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2.5</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data_tmp</span><span class="o">$</span><span class="n">study</span><span class="p">))</span>
<span class="w">  </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">output</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">height</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">width</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="p">}</span>

<span class="nf">saveRDS</span><span class="p">(</span><span class="n">enrichment_data_meta</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO/EOO_all_enrichment_withmeta.rds&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;eQTL&quot;
[1] &quot;multi_context&quot;
[1] &quot;Other&quot;
[1] &quot;snuc_eQTL&quot;
[1] &quot;pQTL&quot;
[1] &quot;multi_gene&quot;
[1] &quot;sQTL&quot;
[1] &quot;Epi&quot;
[1] &quot;metabolome&quot;
[1] &quot;all&quot;
[1] &quot;eQTL&quot;
[1] &quot;multi_context&quot;
[1] &quot;Other&quot;
[1] &quot;snuc_eQTL&quot;
[1] &quot;pQTL&quot;
[1] &quot;multi_gene&quot;
[1] &quot;sQTL&quot;
[1] &quot;Epi&quot;
[1] &quot;metabolome&quot;
[1] &quot;all&quot;
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="test-between-methods">
<h3>test between methods<a class="headerlink" href="#test-between-methods" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># test data </span>

<span class="c1"># To identify annotations with significantly different enrichment values between the single-context (eQTL) and multi-gene (adj_eQTL) datasets, </span>
<span class="c1"># a statistical test was performed on annotations with significant enrichment (Enrichment_P_value &lt; 0.05) in at least one dataset. </span>
<span class="c1"># The datasets were merged by matching annotations, retaining only those significant in either dataset, and rows with missing or infinite values were excluded. </span>
<span class="c1"># Due to the non-normal distribution of enrichment differences (confirmed by a Shapiro-Wilk p-value of 5.623531e-60 in one context), </span>
<span class="c1"># a Z-score test was applied to each paired annotation, calculating the difference (Enrichment_adj - Enrichment_pqtl) divided by the combined standard error.</span>
<span class="c1"># P-values were derived from Z-scores and adjusted for multiple testing using the False Discovery Rate (FDR) method. </span>
<span class="c1"># Annotations with FDR-adjusted p-values &lt; 0.05 were considered significantly different, </span>
<span class="c1"># indicating systematic differences in enrichment between the single-context and multi-gene datasets for those annotations.</span>
</pre></div>
</div>
</div>
</div>
<section id="multi-gene-to-single-context">
<h4>multi gene to single context<a class="headerlink" href="#multi-gene-to-single-context" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">bulk_eQTL</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;eQTL&#39;</span><span class="p">)</span>
<span class="n">snuc_eQTL</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;snuc_eQTL&#39;</span><span class="p">)</span>
<span class="n">eQTL_raw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">bulk_eQTL</span><span class="p">,</span><span class="w"> </span><span class="n">snuc_eQTL</span><span class="p">)</span>
<span class="n">multi_gene_eQTL</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;multi_gene&#39;</span><span class="p">)</span>

<span class="c1"># build a modalities dict to combine single context and multi gene results </span>
<span class="n">modality_dict</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span>
<span class="w">  </span><span class="c1"># MSBB</span>
<span class="w">  </span><span class="s">&quot;BM_10_MSBB_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;MSBB_BM_10_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;BM_22_MSBB_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;MSBB_BM_22_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;BM_36_MSBB_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;MSBB_BM_36_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;BM_44_MSBB_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;MSBB_BM_44_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># DeJager and mega (ROSMAP)</span>
<span class="w">  </span><span class="s">&quot;AC_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_AC_DeJager_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Ast_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Ast_DeJager_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Ast_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Ast_mega_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;DLPFC_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_DLPFC_DeJager_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Exc_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Exc_DeJager_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Exc_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Exc_mega_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Inh_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Inh_DeJager_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Inh_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Inh_mega_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Mic_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Mic_DeJager_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Mic_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Mic_mega_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;monocyte_ROSMAP_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_monocyte_ROSMAP_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Oli_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Oli_DeJager_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Oli_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Oli_mega_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;OPC_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_OPC_DeJager_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;OPC_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_OPC_mega_eQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;PCC_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_PCC_DeJager_eQTL&quot;</span>
<span class="p">)</span>
<span class="n">eQTL</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">eQTL_raw</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">modality_dict</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">recode</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="o">!!!</span><span class="n">modality_dict</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="w"> </span>
<span class="n">contexts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">modality_dict</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">as.character</span><span class="p">()</span>

<span class="n">combined_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map_dfr</span><span class="p">(</span><span class="n">contexts</span><span class="p">,</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">context</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1"># Step 1: Filter significant annotations (Enrichment_P_value &lt; 0.05 in at least one dataset)</span>
<span class="w">  </span><span class="n">significant_single</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">eQTL</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="n">Annotation</span><span class="p">)</span>
<span class="w">  </span><span class="n">significant_multi</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_gene_eQTL</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="n">Annotation</span><span class="p">)</span>
<span class="w">  </span><span class="n">significant_annotations</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">significant_single</span><span class="o">$</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">significant_multi</span><span class="o">$</span><span class="n">Annotation</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 2: Merge datasets for significant annotations</span>
<span class="w">  </span><span class="n">merged_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">inner_join</span><span class="p">(</span>
<span class="w">    </span><span class="n">eQTL</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">significant_annotations</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">select</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_single_context</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_SE_single_context</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_SE</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value_single_context</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_P_value</span><span class="p">),</span>
<span class="w">    </span><span class="n">multi_gene_eQTL</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">significant_annotations</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">select</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_multi_gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_SE_multi_gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_SE</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value_multi_gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_P_value</span><span class="p">),</span>
<span class="w">    </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Annotation&quot;</span>
<span class="w">  </span><span class="p">)</span><span class="w">  </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="nf">if_all</span><span class="p">(</span><span class="nf">starts_with</span><span class="p">(</span><span class="s">&quot;Enrichment&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">is.finite</span><span class="p">))</span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;_OTHER&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">      </span><span class="c1"># Annotation = gsub(&#39;.bed.gz&#39;, &#39;&#39;, Annotation), </span>
<span class="w">           </span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;.bed&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;formatted_output_&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">.</span><span class="p">)</span><span class="w"> </span><span class="p">,</span>
<span class="w">           </span><span class="c1"># , </span>
<span class="w">           </span><span class="c1"># Annotation = gsub(&#39;Transcription_Factor_&#39;, &#39;TF_&#39;, Annotation), </span>
<span class="w">           </span><span class="c1"># Annotation = gsub(&#39;Other_Functional_Annotations&#39;, &#39;Others&#39;, Annotation), </span>
<span class="w">           </span><span class="c1"># Annotation = gsub(&#39;Chromatin_Interaction_Loops&#39;, &#39;CILs&#39;, Annotation), </span>
<span class="w">           </span><span class="c1"># Annotation = gsub(&#39;.HuFNSC02.narrowPeak.lifted&#39;, &#39;&#39;, Annotation), </span>
<span class="w">           </span><span class="c1"># Annotation = gsub(&#39;_lifted&#39;, &#39;&#39;, Annotation), </span>
<span class="w">           </span><span class="c1"># Annotation = gsub(&#39;UCSF−UBC.&#39;, &#39;&#39;, Annotation),  </span>
<span class="w">           </span><span class="c1"># Annotation = gsub(&#39;_hg38lift.&#39;, &#39;&#39;, Annotation), </span>
<span class="w">           </span><span class="c1"># Annotation = gsub(&#39;_SuperEnhancer&#39;, &#39;&#39;, Annotation), </span>
<span class="w">           </span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;_OTHER_significant&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="p">)</span>
<span class="w">           </span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 3: Check if there are valid paired observations</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">merged_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;No valid paired observations for context: &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">context</span><span class="p">)</span>
<span class="w">    </span><span class="kr">return</span><span class="p">(</span><span class="kc">NULL</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 4: Perform Z-score test and calculate differences</span>
<span class="w">  </span><span class="n">results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merged_data</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">      </span><span class="n">Context</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">context</span><span class="p">,</span>
<span class="w">      </span><span class="n">Enrichment_Diff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_multi_gene</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">Enrichment_single_context</span><span class="p">,</span>
<span class="w">      </span><span class="n">Enrichment_SE_combined</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sqrt</span><span class="p">(</span><span class="n">Enrichment_SE_single_context</span><span class="o">^</span><span class="m">2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">Enrichment_SE_multi_gene</span><span class="o">^</span><span class="m">2</span><span class="p">),</span>
<span class="w">      </span><span class="n">z_score</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_Diff</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">Enrichment_SE_combined</span><span class="p">,</span>
<span class="w">      </span><span class="n">p_value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">pnorm</span><span class="p">(</span><span class="o">-</span><span class="nf">abs</span><span class="p">(</span><span class="n">z_score</span><span class="p">)),</span>
<span class="w">      </span><span class="n">p_adjusted</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">p.adjust</span><span class="p">(</span><span class="n">p_value</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;fdr&quot;</span><span class="p">),</span>
<span class="w">      </span><span class="n">Significance_Status</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">        </span><span class="n">Enrichment_P_value_single_context</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">Enrichment_P_value_multi_gene</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;Both&quot;</span><span class="p">,</span>
<span class="w">        </span><span class="n">Enrichment_P_value_single_context</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;Single Context Only&quot;</span><span class="p">,</span>
<span class="w">        </span><span class="n">Enrichment_P_value_multi_gene</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;Multi-Gene Only&quot;</span>
<span class="w">      </span><span class="p">)</span>
<span class="w">    </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="n">Context</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_single_context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_multi_gene</span><span class="p">,</span>
<span class="w">           </span><span class="n">Enrichment_Diff</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_SE_single_context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_SE_multi_gene</span><span class="p">,</span>
<span class="w">           </span><span class="n">Enrichment_P_value_single_context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value_multi_gene</span><span class="p">,</span><span class="w"> </span><span class="n">Significance_Status</span><span class="p">,</span>
<span class="w">           </span><span class="n">p_value</span><span class="p">,</span><span class="w"> </span><span class="n">p_adjusted</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 5: Filter significant results (FDR &lt; 0.05)</span>
<span class="w">  </span><span class="n">significant</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">results</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">p_adjusted</span><span class="p">),</span><span class="w"> </span><span class="n">p_adjusted</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\\|&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)[,</span><span class="w"> </span><span class="m">1</span><span class="p">])</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">.keep_all</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">significant</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">significant</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">Enrichment_Diff</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annotation</span><span class="p">))</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">plot_data</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_Diff</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Category</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;identity&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dashed&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">labs</span><span class="p">(</span>
<span class="w">        </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;Top Enrichment Differences for &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">context</span><span class="p">),</span>
<span class="w">        </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Enrichment Difference (multi_gene - single_context)&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span>
<span class="w">      </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">all_colors</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">out_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO//enrichment_figures_category&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;methods_compare/multi_gene/&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;_multigene_single_enrichment_plot_combined.pdf&#39;</span><span class="p">))</span>
<span class="w">    </span><span class="nf">dir.create</span><span class="p">(</span><span class="nf">dirname</span><span class="p">(</span><span class="n">out_path</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">    </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">out_path</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">14</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span>
<span class="w">  </span><span class="n">results</span>
<span class="p">})</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 209 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 407 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 112 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 87 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 17 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 698 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 100 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 69 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 50 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 330 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 2 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 231 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 92 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 1035 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 48 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 889 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 162 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 14 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 19 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 688 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 29 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 107 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 35 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 678 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 44 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 330 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 188 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 371 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 87 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 758 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 109 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 167 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 77 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 38 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 33 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 347 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 27 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 1140 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 23 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 829 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">significant_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">combined_results</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">p_adjusted</span><span class="p">),</span><span class="w"> </span><span class="n">p_adjusted</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\\|&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)[,</span><span class="w"> </span><span class="m">1</span><span class="p">])</span>

<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Number of significantly different annotations (FDR &lt; 0.05):&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">significant_results</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>

<span class="n">top_num</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2</span>
<span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">significant_results</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">Context</span><span class="p">,</span><span class="w"> </span><span class="n">Category</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">Enrichment_Diff</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">top_num</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">Annotation</span><span class="p">)))</span>

<span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">plot_data</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_Diff</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_Diff</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;identity&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;skyblue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;salmon&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">guide</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;none&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dashed&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span>
<span class="w">    </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Top Enrichment Differences for Significant Annotations (FDR &lt; 0.05)&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Enrichment Difference (multi_gene - single_context)&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">Context</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;free_y&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">),</span><span class="w"> </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">))</span>

<span class="n">out_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO//enrichment_figures_category&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;methods_compare/multi_gene/&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;all_multigene_single_enrichment_plot_combined.pdf&#39;</span><span class="p">)</span>
<span class="nf">dir.create</span><span class="p">(</span><span class="nf">dirname</span><span class="p">(</span><span class="n">out_path</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">out_path</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.5</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">Context</span><span class="p">)),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">60</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of significantly different annotations (FDR &lt; 0.05): 3414 
</pre></div>
</div>
</div>
</div>
</section>
<section id="multi-context-to-single-context">
<h4>multi context to single context<a class="headerlink" href="#multi-context-to-single-context" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">enrichment_data_meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;multi_context&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">study</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'AC_DeJager_eQTL_multicontext'</li><li>'Ast_DeJager_eQTL_multicontext'</li><li>'Ast_Kellis_eQTL_multicontext'</li><li>'Ast_mega_eQTL_multicontext'</li><li>'DLPFC_Bennett_pQTL_multicontext'</li><li>'DLPFC_DeJager_eQTL_multicontext'</li><li>'Exc_DeJager_eQTL_multicontext'</li><li>'Exc_Kellis_eQTL_multicontext'</li><li>'Exc_mega_eQTL_multicontext'</li><li>'Inh_DeJager_eQTL_multicontext'</li><li>'Inh_Kellis_eQTL_multicontext'</li><li>'Inh_mega_eQTL_multicontext'</li><li>'Mic_DeJager_eQTL_multicontext'</li><li>'Mic_Kellis_eQTL_multicontext'</li><li>'Mic_mega_eQTL_multicontext'</li><li>'monocyte_ROSMAP_eQTL_multicontext'</li><li>'Oli_DeJager_eQTL_multicontext'</li><li>'Oli_Kellis_eQTL_multicontext'</li><li>'Oli_mega_eQTL_multicontext'</li><li>'OPC_DeJager_eQTL_multicontext'</li><li>'OPC_Kellis_eQTL_multicontext'</li><li>'OPC_mega_eQTL_multicontext'</li><li>'PCC_DeJager_eQTL_multicontext'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Filter eQTL data</span>
<span class="n">bulk_eQTL</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;eQTL&#39;</span><span class="p">)</span>
<span class="n">snuc_eQTL</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;snuc_eQTL&#39;</span><span class="p">)</span>
<span class="n">eQTL_raw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">bulk_eQTL</span><span class="p">,</span><span class="w"> </span><span class="n">snuc_eQTL</span><span class="p">)</span>
<span class="n">multi_context_eQTL</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_data_meta</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">`Data Type`</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;multi_context&#39;</span><span class="p">)</span>

<span class="c1"># Build a modalities dict to combine single context and multi gene results</span>
<span class="n">modality_dict_multicontext</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span>
<span class="w">  </span><span class="s">&quot;AC_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;AC_DeJager_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;DLPFC_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DLPFC_DeJager_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;PCC_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;PCC_DeJager_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Ast_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Ast_DeJager_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Inh_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Inh_DeJager_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Exc_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Exc_DeJager_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Mic_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mic_DeJager_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;OPC_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;OPC_DeJager_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Oli_DeJager_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Oli_DeJager_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;PCC_Kellis_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;PCC_Kellis_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Ast_Kellis_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Ast_Kellis_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Inh_Kellis_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Inh_Kellis_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Exc_Kellis_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Exc_Kellis_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Mic_Kellis_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mic_Kellis_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;OPC_Kellis_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;OPC_Kellis_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Oli_Kellis_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Oli_Kellis_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;PCC_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;PCC_mega_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Ast_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Ast_mega_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Inh_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Inh_mega_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Exc_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Exc_mega_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Mic_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mic_mega_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;OPC_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;OPC_mega_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Oli_mega_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Oli_mega_eQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;DLPFC_Bennett_pQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DLPFC_Bennett_pQTL_multicontext&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;monocyte_ROSMAP_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;monocyte_ROSMAP_eQTL_multicontext&quot;</span>

<span class="p">)</span>

<span class="c1"># Harmonize study names</span>
<span class="n">eQTL</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">eQTL_raw</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">modality_dict_multicontext</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">recode</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="o">!!!</span><span class="n">modality_dict_multicontext</span><span class="p">))</span>

<span class="n">multi_context_eQTL</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_context_eQTL</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">modality_dict_multicontext</span><span class="p">))</span>

<span class="c1"># Compare enrichment across modalities</span>
<span class="n">contexts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">modality_dict_multicontext</span><span class="p">)</span>
<span class="n">combined_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map_dfr</span><span class="p">(</span><span class="n">contexts</span><span class="p">,</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">context</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w"> </span>
<span class="w">  </span><span class="n">significant_single</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">eQTL</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="n">Annotation</span><span class="p">)</span><span class="w"> </span>
<span class="w">  </span><span class="n">significant_multi</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_context_eQTL</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="n">Annotation</span><span class="p">)</span><span class="w"> </span>
<span class="w">  </span><span class="n">significant_annotations</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">significant_single</span><span class="o">$</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">significant_multi</span><span class="o">$</span><span class="n">Annotation</span><span class="p">))</span><span class="w"> </span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 2: Merge datasets for significant annotations </span>
<span class="w">  </span><span class="n">merged_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">inner_join</span><span class="p">(</span><span class="w"> </span>
<span class="w">    </span><span class="n">eQTL</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">significant_annotations</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">select</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_single_context</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_SE_single_context</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_SE</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value_single_context</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_P_value</span><span class="p">),</span><span class="w"> </span>
<span class="w">    </span><span class="n">multi_context_eQTL</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">filter</span><span class="p">(</span><span class="n">study</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">significant_annotations</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">select</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_multi_gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_SE_multi_gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_SE</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value_multi_gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_P_value</span><span class="p">),</span><span class="w"> </span>
<span class="w">    </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Annotation&quot;</span><span class="w"> </span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="nf">if_all</span><span class="p">(</span><span class="nf">starts_with</span><span class="p">(</span><span class="s">&quot;Enrichment&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">is.finite</span><span class="p">))</span><span class="w"> </span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 3: Check if there are valid paired observations </span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">merged_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w"> </span>
<span class="w">    </span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;No valid paired observations for context: &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">context</span><span class="p">)</span><span class="w"> </span>
<span class="w">    </span><span class="kr">return</span><span class="p">(</span><span class="kc">NULL</span><span class="p">)</span><span class="w"> </span>
<span class="w">  </span><span class="p">}</span><span class="w"> </span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 4: Perform Z-score test and calculate differences </span>
<span class="w">  </span><span class="n">results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merged_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="w"> </span>
<span class="w">      </span><span class="n">Context</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">context</span><span class="p">,</span><span class="w"> </span>
<span class="w">      </span><span class="n">Enrichment_Diff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_multi_gene</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">Enrichment_single_context</span><span class="p">,</span><span class="w"> </span>
<span class="w">      </span><span class="n">Enrichment_SE_combined</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sqrt</span><span class="p">(</span><span class="n">Enrichment_SE_single_context</span><span class="o">^</span><span class="m">2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">Enrichment_SE_multi_gene</span><span class="o">^</span><span class="m">2</span><span class="p">),</span><span class="w"> </span>
<span class="w">      </span><span class="n">z_score</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_Diff</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">Enrichment_SE_combined</span><span class="p">,</span><span class="w"> </span>
<span class="w">      </span><span class="n">p_value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">pnorm</span><span class="p">(</span><span class="o">-</span><span class="nf">abs</span><span class="p">(</span><span class="n">z_score</span><span class="p">)),</span><span class="w"> </span>
<span class="w">      </span><span class="n">p_adjusted</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">p.adjust</span><span class="p">(</span><span class="n">p_value</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;fdr&quot;</span><span class="p">),</span><span class="w"> </span>
<span class="w">      </span><span class="n">Significance_Status</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">case_when</span><span class="p">(</span><span class="w"> </span>
<span class="w">        </span><span class="n">Enrichment_P_value_single_context</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">Enrichment_P_value_multi_gene</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;Both&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">        </span><span class="n">Enrichment_P_value_single_context</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;Single Context Only&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">        </span><span class="n">Enrichment_P_value_multi_gene</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;Multi-Gene Only&quot;</span><span class="w"> </span>
<span class="w">      </span><span class="p">)</span><span class="w"> </span>
<span class="w">    </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="n">Context</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_single_context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_multi_gene</span><span class="p">,</span><span class="w"> </span>
<span class="w">           </span><span class="n">Enrichment_Diff</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_SE_single_context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_SE_multi_gene</span><span class="p">,</span><span class="w"> </span>
<span class="w">           </span><span class="n">Enrichment_P_value_single_context</span><span class="p">,</span><span class="w"> </span><span class="n">Enrichment_P_value_multi_gene</span><span class="p">,</span><span class="w"> </span><span class="n">Significance_Status</span><span class="p">,</span><span class="w"> </span>
<span class="w">           </span><span class="n">p_value</span><span class="p">,</span><span class="w"> </span><span class="n">p_adjusted</span><span class="p">)</span><span class="w"> </span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 5: Filter significant results (FDR &lt; 0.05) </span>
<span class="w">  </span><span class="n">significant</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">results</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">p_adjusted</span><span class="p">),</span><span class="w"> </span><span class="n">p_adjusted</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&quot;\\|&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">Annotation</span><span class="p">),</span><span class="w"> </span>
<span class="w">                             </span><span class="nf">str_split</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\\|&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span><span class="w"> </span>
<span class="w">                             </span><span class="n">Annotation</span><span class="p">))</span><span class="w">  </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">.keep_all</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>

<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">significant</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w"> </span>
<span class="w">    </span><span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">significant</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">Enrichment_Diff</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">      </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annotation</span><span class="p">))</span><span class="w"> </span>
<span class="w">    </span>
<span class="w">    </span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">plot_data</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_Diff</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Category</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;identity&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dashed&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">labs</span><span class="p">(</span><span class="w"> </span>
<span class="w">        </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;Top Enrichment Differences for &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">context</span><span class="p">),</span><span class="w"> </span>
<span class="w">        </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Enrichment Difference (multi_context - single_context)&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="w"> </span>
<span class="w">      </span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">all_colors</span><span class="p">)</span><span class="w"> </span>
<span class="w">    </span>
<span class="w">    </span><span class="n">out_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO//enrichment_figures_category&#39;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                          </span><span class="s">&#39;methods_compare/multi_context/&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">context</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;_multigene_single_enrichment_plot_combined.pdf&#39;</span><span class="p">))</span><span class="w"> </span>
<span class="w">    </span><span class="nf">dir.create</span><span class="p">(</span><span class="nf">dirname</span><span class="p">(</span><span class="n">out_path</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span>
<span class="w">    </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">out_path</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">14</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span>
<span class="w">  </span><span class="p">}</span><span class="w"> </span>
<span class="w">  </span>
<span class="w">  </span><span class="n">results</span><span class="w"> </span>
<span class="p">})</span>

<span class="c1"># Filter for significant results and plot summary</span>
<span class="n">significant_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">combined_results</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">p_adjusted</span><span class="p">),</span><span class="w"> </span><span class="n">p_adjusted</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_extract</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;^[^|]+&quot;</span><span class="p">))</span>

<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Number of significantly different annotations (FDR &lt; 0.05):&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">significant_results</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>

<span class="n">top_num</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2</span>
<span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">significant_results</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">Context</span><span class="p">,</span><span class="w"> </span><span class="n">Category</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">Enrichment_Diff</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">top_num</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">Annotation</span><span class="p">)))</span>

<span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">plot_data</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_Diff</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annotation</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Enrichment_Diff</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;identity&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;skyblue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;salmon&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">guide</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;none&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dashed&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span>
<span class="w">    </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Top Enrichment Differences for Significant Annotations (FDR &lt; 0.05)&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Enrichment Difference (multi_context - single_context)&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">Context</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;free_y&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">),</span><span class="w"> </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">))</span>

<span class="n">out_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="s">&#39;~/Work/Fungen_xQTL/2025/Sep/eoo/EOO//enrichment_figures_category&#39;</span><span class="p">,</span>
<span class="w">                      </span><span class="s">&#39;methods_compare/multi_context/&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;all_multigene_single_enrichment_plot_combined.pdf&#39;</span><span class="p">)</span>
<span class="nf">dir.create</span><span class="p">(</span><span class="nf">dirname</span><span class="p">(</span><span class="n">out_path</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">out_path</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2.3</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">Context</span><span class="p">)),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 51 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 1349 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 49 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 218 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 25 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 465 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 2 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 920 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 33 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 780 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 32 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 220 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 43 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 637 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 37 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 486 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 104 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 744 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
No valid paired observations for context: PCC_Kellis_eQTL_multicontext

Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 294 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 509 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 148 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 717 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 18 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 149 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 13 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 157 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 5 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 79 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 18 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 193 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
No valid paired observations for context: PCC_mega_eQTL_multicontext

Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 21 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 486 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 39 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 435 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 21 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 403 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 109 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 484 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 27 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 910 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 80 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 1157 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
No valid paired observations for context: DLPFC_Bennett_pQTL_multicontext

Warning message in inner_join(eQTL %&gt;% filter(study == context, Annotation %in% :
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 20 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 844 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of significantly different annotations (FDR &lt; 0.05): 2487 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">ggsave</span><span class="p">(</span><span class="n">out_path</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.3</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">Context</span><span class="p">)),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">50</span><span class="p">,</span><span class="w"> </span><span class="n">limitsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="summary-of-updated-multi-context-eoo-results">
<h2>Summary of Updated Multi-Context EOO Results<a class="headerlink" href="#summary-of-updated-multi-context-eoo-results" title="Link to this heading">#</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>1.	The updated results include more contexts than the previous version (Apr 2025).
2.	Overall patterns remain similar, especially when comparing across AC.
3.  Subset of annotations shows significant differences in enrichment between the multi-context and single-context models (FDR &lt; 0.05).
4.	The magnitude and direction of enrichment differences vary across annotations.
5.	Some annotations (e.g., PU1 promoter/enhancer in microglia) show higher enrichment in the multi-context model.
6.	Others (e.g., LHX2 promoter in astrocytes) show higher enrichment in the single-context model.
7.	The pattern of differences is not consistent across cell types or annotation categories.
</pre></div>
</div>
<p>The visualization should be replicated starting from Viusalization - Plot; input data has been uploaded to s3://statfungen/ftp_fgc_xqtl/analysis_result/EOO/EOO_reshape_combined_result_withSNP_pip01_95cs_annotation_variant_level.updatedSep9th.RDS</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./main_text/2_single_context_cis/staging"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#input">input</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-preparation">Data preparation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization">Visualization</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#prepare-data">prepare data</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#load-updated-multicontext">load updated multicontext</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#load-all-other-data">load all other data</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#combine-as-new-data">combine as new data</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plot">Plot</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#barplot">barplot</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#volcano">volcano</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#test-between-methods">test between methods</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-to-single-context">multi gene to single context</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-context-to-single-context">multi context to single context</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#summary-of-updated-multi-context-eoo-results">Summary of Updated Multi-Context EOO Results</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>