
<!DOCTYPE html>


<html lang="en" data-content_root="../../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>SLDSC &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/2_single_context_cis/staging/sldsc_plots';</script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../1_project_overview/index.html">Multi-omics QTL atlas of the aging human brain</a></li>
<li class="toctree-l1"><a class="reference internal" href="../index.html">Fine-mapped QTL landscape across brain molecular phenotypes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../3_single_context_multigene_trans/index.html">Coordinated gene regulation and long-range genetic effects</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../4_multi_context/index.html">Cascading genetic effects across molecular contexts</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../5_AD_xQTL_integration/index.html">Regulatory architecture of AD susceptibility loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5a_AD_GWAS_loci_harmonized.html">AD GWAS loci harmonization and refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5b_AD_GWAS_loci_properties.html">Refined AD GWAS loci properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5c_AD_loci_xQTL_annotated.html">AD loci “explained” by xQTL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5d_AD_loci_table_summary.html">Summary of FunGen-xQTL AD loci table</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5e_GSEA.html">Gene Set Enrichment Analysis for AD Risk Genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5f_PPI.html">PPI Networks for Selected AD Risk Genes in xQTL Loci</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5g_cV2F.html">Consensus Variant-to-Function Score for AD</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../6_AD_xQTL_genes/index.html">Molecular dissection of prioritized AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.html">Visualization of FunGen-xQTL AD loci table</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../../7_AD_gene_prioritized/index.html">Molecular insights and gene prioritization at key AD loci</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/2_single_context_cis/staging/sldsc_plots.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../../_sources/main_text/2_single_context_cis/staging/sldsc_plots.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>SLDSC</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#plot">Plot</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#meta-traits">Meta traits</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#seprate-traits">Seprate traits</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="sldsc">
<h1>SLDSC<a class="headerlink" href="#sldsc" title="Link to this heading">#</a></h1>
<section id="plot">
<h2>Plot<a class="headerlink" href="#plot" title="Link to this heading">#</a></h2>
<section id="meta-traits">
<h3>Meta traits<a class="headerlink" href="#meta-traits" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>## Library
library(data.table)
library(ggplot2)
library(RColorBrewer)
library(colorspace)
library(ggpubr)
library(tidyverse)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>studies &lt;- c(&#39;ROSMAP_gpQTL_glycosylation_unadjusted&#39;) ## change context
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>res.dir &lt;- &quot;./&quot; ## change path

tau.list &lt;- list()
enrich.list &lt;- list()
for (study in studies) {
  meta.dir &lt;- paste0(res.dir, study, &quot;/sumstats/all_m/processed_stats_2/single_tau.&quot;,study, &quot;.meta_processed_stats.rds&quot;)
  metas &lt;- readRDS(meta.dir)
  tau.list[[study]] &lt;- data.frame(metas$tau)
  tau.list[[study]]$study &lt;- study
  tau.list[[study]]$trait &lt;- rownames(tau.list[[study]])
  enrich.list[[study]] &lt;- data.frame(metas$enrichment)
  enrich.list[[study]]$study &lt;- study
  enrich.list[[study]]$trait &lt;- rownames(enrich.list[[study]])
  
}
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>
tau.df &lt;- do.call(rbind, tau.list)
enrich.df &lt;- do.call(rbind, enrich.list)

# 95% confidence interval
tau.df$Lower &lt;- tau.df$Mean - 1.96*tau.df$SD
tau.df$Upper &lt;- tau.df$Mean + 1.96*tau.df$SD
tau.df$signif &lt;- tau.df$P &lt; 0.05
enrich.df$Lower &lt;- enrich.df$Mean - 1.96*enrich.df$SD
enrich.df$Upper &lt;- enrich.df$Mean + 1.96*enrich.df$SD
enrich.df$signif &lt;- enrich.df$P &lt; 0.05

</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>reorder_levels &lt;- tau.df %&gt;%
  filter(trait == &quot;Brain&quot;) %&gt;%          # choose reference trait
  arrange(desc(Mean)) %&gt;%               # descending by Mean
  pull(study) %&gt;%                       # get vector
  unique()

tau.df$study &lt;- factor(tau.df$study, levels = reorder_levels)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>## 5. Mark significance (95 % CI does NOT include zero) ---------
tau.df &lt;- tau.df %&gt;%
  mutate(signif = Lower * Upper &gt; 0)

## 6. Prepare a colour palette ---------------------------------
all_colors &lt;- qualitative_hcl(
  n = length(levels(tau.df$study)),
  palette = &quot;Dark 3&quot;
)

## 7. Plot marginal_tau -------------------------------------------------------
p &lt;- ggplot(
  tau.df,
  aes(x = study, y = Mean,
      colour = study, group = trait)         
) +
  # point estimates
  geom_point(
    position = position_dodge(width = 0.6),
    size = 2
  ) +
  # error bars (95 % CI)
  geom_errorbar(
    aes(ymin = Lower, ymax = Upper,
        linewidth = signif),               # thick if significant
    position = position_dodge(width = 0.6),
    width = 0.25
  ) +
  # asterisks on significant estimates
  geom_text(
    data = subset(tau.df, signif),
    aes(label = &quot;*&quot;, y = Upper + 0.05),
    position = position_dodge(width = 0.6),
    colour = &quot;red&quot;, size = 4
  ) +
  # colour &amp; linewidth scales
  scale_color_manual(values = all_colors) +
  scale_linewidth_manual(values = c(`TRUE` = 1.2, `FALSE` = 0.5)) +
  # horizontal zero line
  geom_hline(yintercept = 0, linetype = 2, colour = &quot;darkblue&quot;) +
  # theme tweaks
  theme_bw() +
  theme(
    axis.text.x  = element_text(angle = 45, hjust = 1, vjust = 1),
    axis.text.y  = element_text(size = 9,  colour = &quot;black&quot;),
    axis.title.x = element_blank(),
    legend.position = &quot;none&quot;
  ) +
  # axis title
  ylab(&quot;Tau (Mean ± 1.96 × SD)&quot;) + facet_wrap(~trait)

## 8. Display plot ----------------------------------------------
print(p)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>## Enrichment

reorder_levels &lt;- enrich.df %&gt;%
  filter(trait == &quot;Brain&quot;) %&gt;%          # choose reference trait
  arrange(desc(Mean)) %&gt;%               # descending by Mean
  pull(study) %&gt;%                       # get vector
  unique()

enrich.df$study &lt;- factor(enrich.df$study, levels = reorder_levels)

## 5. Mark significance (95 % CI does NOT include zero) ---------
enrich.df &lt;- enrich.df %&gt;%
  mutate(signif = Lower * Upper &gt; 0)

## 6. Prepare a colour palette ---------------------------------
all_colors &lt;- qualitative_hcl(
  n = length(levels(enrich.df$study)),
  palette = &quot;Dark 3&quot;
)

## 7. Plot Enrichment-------------------------------------------------------
p &lt;- ggplot(
  enrich.df,
  aes(x = study, y = Mean,
      colour = study)      
) +
  # point estimates
  geom_point(
    position = position_dodge(width = 0.6),
    size = 2
  ) +
  # error bars (95 % CI)
  geom_errorbar(
    aes(ymin = Lower, ymax = Upper,
        linewidth = signif),               # thick if significant
    position = position_dodge(width = 0.6),
    width = 0.25
  ) +
  # asterisks on significant estimates
  geom_text(
    data = subset(enrich.df, signif),
    aes(label = &quot;*&quot;, y = Upper + 2),
    position = position_dodge(width = 0.6),
    colour = &quot;red&quot;, size = 4
  ) +
  # colour &amp; linewidth scales
  scale_color_manual(values = all_colors) +
  scale_linewidth_manual(values = c(`TRUE` = 1.2, `FALSE` = 0.5)) +
  # horizontal zero line
  geom_hline(yintercept = 1, linetype = 2, colour = &quot;darkblue&quot;) +
  # theme tweaks
  theme_bw() +
  theme(
    axis.text.x  = element_text(angle = 45, hjust = 1, vjust = 1),
    axis.text.y  = element_text(size = 9,  colour = &quot;black&quot;),
    axis.title.x = element_blank(),
    legend.position = &quot;none&quot;
  ) +
  # axis title
  ylab(&quot;Enrichment (Mean ± 1.96 × SD)&quot;) + facet_wrap(~trait)



## 8. Display plot ----------------------------------------------
print(p)
</pre></div>
</div>
</div>
</div>
</section>
<section id="seprate-traits">
<h3>Seprate traits<a class="headerlink" href="#seprate-traits" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(tidyverse)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>## Library
library(data.table)
library(ggplot2)
library(RColorBrewer)
library(colorspace)
library(ggpubr)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>ttmp &lt;- readRDS(&#39;./ROSMAP_gpQTL_glycosylation_unadjusted.single_tau.initial_processed_stats.rds&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>blood_traits &lt;- c(&#39;UKB.Lym.BOLT.sumstats.parquet&#39;,&#39;UKB.Mono.BOLT.sumstats.parquet&#39;,&#39;UKB.Plt.BOLT.sumstats.parquet&#39;,&#39;UKB.RBC.BOLT.sumstats.parquet&#39;,&#39;UKB_460K.blood_RBC_DISTRIB_WIDTH.sumstats.parquet&#39;,&#39;UKB.Eosino.BOLT.sumstats.parquet&#39;,&#39;UKB.Baso.BOLT.sumstats.parquet&#39;,&#39;UKB.Neutro.BOLT.sumstats.parquet&#39;,&#39;UKB.MCV.BOLT.sumstats.parquet&#39;,&#39;PASS_IBD_deLange2017.sumstats.parquet&#39;,&#39;PASS_CD_deLange2017.sumstats.parquet&#39;,&#39;PASS_UC_deLange2017.sumstats.parquet&#39;,&#39;UKB_460K.disease_ALLERGY_ECZEMA_DIAGNOSED.sumstats.parquet&#39;,&#39;UKB.Hypothyroidism.SAIGE.sumstats.parquet&#39;,&#39;PASS_Rheumatoid_Arthritis.sumstats.parquet&#39;,&#39;PASS_Primary_biliary_cirrhosis.sumstats.parquet&#39;,&#39;PASS_Lupus.sumstats.parquet&#39;,&#39;PASS_Type_1_Diabetes.sumstats.parquet&#39;,&#39;UKB.AID_Combined.SAIGE.sumstats.parquet&#39;,&#39;PASS_Celiac.sumstats.parquet&#39;)
brain_traits &lt;- c(&#39;AD_Bellenguez_buildGRCh38_munged.parquet&#39;, &#39;AD_Wightman_Excluding23andMe_hg38_sorted_munged.parquet&#39;, &#39;AD_Bellenguez_EADB_hg38_sorted_munged.parquet&#39;, &#39;AD_Wightman_ExcludingUKB23andMe_hg38_sorted_munged.parquet&#39;,&#39;AD_Kunkle_etal_Stage1_results_hg38.sorted.munged.parquet&#39;, &#39;AD_Wightman_full_2021_munged.parquet&#39;, &#39;PASS_Alzheimers_Jansen2019.sumstats.parquet&#39;,&#39;EXF.sumstats.parquet&#39;,&#39;LAN.sumstats.parquet&#39;,&#39;MEM.sumstats.parquet&#39;,&#39;MS.sumstats.parquet&#39;,&#39;accumbens.sumstats.parquet&#39;,&#39;ALS-Rheenen-2021.GCST90027163_buildGRCh37.sumstats.parquet&#39;,&#39;ALS-Rheenen-2021.GCST90027164_buildGRCh37.sumstats.parquet&#39;,&#39;amygdala.sumstats.parquet&#39;,&#39;brainstem.sumstats.parquet&#39;,&#39;caudate.sumstats.parquet&#39;,&#39;nallsEtAl2019_excluding23andMe.sumstats.parquet&#39;,&#39;neuroticism_ctg.sumstats.parquet&#39;,&#39;pallidum.sumstats.parquet&#39;,&#39;PGC3_SCZ_wave3_public.v2.sumstats.parquet&#39;,&#39;PGC_UKB_23andMe_depression_10000.sumstats.parquet&#39;,&#39;PGC_UKB_depression_genome-wide.sumstats.parquet&#39;,&#39;putamen.sumstats.parquet&#39;,&#39;rmf.surf.sumstats.parquet&#39;,&#39;rmf.thick.sumstats.parquet&#39;,&#39;sf.surf.sumstats.parquet&#39;,&#39;sf.thick.sumstats.parquet&#39;,&#39;image_AD1.sumstats.parquet&#39;,&#39;image_Aging2.sumstats.parquet&#39;,&#39;image_Aging5.sumstats.parquet&#39;,&#39;image_AD2.sumstats.parquet&#39;,&#39;image_Aging3.sumstats.parquet&#39;,&#39;image_Aging1.sumstats.parquet&#39;,&#39;image_Aging4.sumstats.parquet&#39;,&#39;UKB_460K.repro_NumberChildrenEverBorn_Pooled.sumstats.parquet&#39;,&#39;PASS_Anorexia.sumstats.parquet&#39;,&#39;PASS_ADHD_Demontis2018.sumstats.parquet&#39;,&#39;PASS_Autism.sumstats.parquet&#39;,&#39;PASS_SleepDuration_Dashti2019.sumstats.parquet&#39;,&#39;UKB.BMI.BOLT.sumstats.parquet&#39;,&#39;PASS_MDD_Wray2018.sumstats.parquet&#39;,&#39;PASS_Neuroticism.sumstats.parquet&#39;,&#39;UKB.Smoking_Ever_Never.SAIGE.sumstats.parquet&#39;,&#39;UKB_460K.cov_EDU_YEARS.sumstats.parquet&#39;,&#39;PASS_Intelligence_SavageJansen2018.sumstats.parquet&#39;,&#39;UKB.Morning_Person.BOLT.sumstats.parquet&#39;,&#39;UKB.Insomnia.BOLT.sumstats.parquet&#39;,&#39;PASS_Schizophrenia.sumstats.parquet&#39;,&#39;PASS_SCZvsBD_Ruderfer2018.sumstats.parquet&#39;,&#39;PASS_BipolarDisorder_Ruderfer2018.sumstats.parquet&#39;,&#39;PASS_ReactionTime_Davies2018.sumstats.parquet&#39;,&#39;PASS_AgeFirstBirth.sumstats.parquet&#39;,&#39;PASS_Multiple_sclerosis.sumstats.parquet&#39;)
all_traits &lt;- c(&#39;AD_Bellenguez_buildGRCh38_munged.parquet&#39;, &#39;AD_Wightman_Excluding23andMe_hg38_sorted_munged.parquet&#39;, &#39;AD_Bellenguez_EADB_hg38_sorted_munged.parquet&#39;, &#39;AD_Wightman_ExcludingUKB23andMe_hg38_sorted_munged.parquet&#39;,&#39;AD_Kunkle_etal_Stage1_results_hg38.sorted.munged.parquet&#39;, &#39;AD_Wightman_full_2021_munged.parquet&#39;, &#39;PASS_Alzheimers_Jansen2019.sumstats.parquet&#39;,&#39;EXF.sumstats.parquet&#39;,&#39;LAN.sumstats.parquet&#39;,&#39;MEM.sumstats.parquet&#39;,&#39;MS.sumstats.parquet&#39;,&#39;accumbens.sumstats.parquet&#39;,&#39;ALS-Rheenen-2021.GCST90027163_buildGRCh37.sumstats.parquet&#39;,&#39;ALS-Rheenen-2021.GCST90027164_buildGRCh37.sumstats.parquet&#39;,&#39;amygdala.sumstats.parquet&#39;,&#39;brainstem.sumstats.parquet&#39;,&#39;caudate.sumstats.parquet&#39;,&#39;GWAS_CP_all.sumstats.parquet&#39;,&#39;GWAS_EA_excl23andMe.sumstats.parquet&#39;,&#39;MegaGWAS_summary_European.sumstats.parquet&#39;,&#39;nallsEtAl2019_excluding23andMe.sumstats.parquet&#39;,&#39;neuroticism_ctg.sumstats.parquet&#39;,&#39;pallidum.sumstats.parquet&#39;,&#39;PGC3_SCZ_wave3_public.v2.sumstats.parquet&#39;,&#39;PGC_UKB_23andMe_depression_10000.sumstats.parquet&#39;,&#39;PGC_UKB_depression_genome-wide.sumstats.parquet&#39;,&#39;putamen.sumstats.parquet&#39;,&#39;rmf.surf.sumstats.parquet&#39;,&#39;rmf.thick.sumstats.parquet&#39;,&#39;sf.surf.sumstats.parquet&#39;,&#39;sf.thick.sumstats.parquet&#39;,&#39;image_AD1.sumstats.parquet&#39;,&#39;image_Aging2.sumstats.parquet&#39;,&#39;image_Aging5.sumstats.parquet&#39;,&#39;image_AD2.sumstats.parquet&#39;,&#39;image_Aging3.sumstats.parquet&#39;,&#39;image_Aging1.sumstats.parquet&#39;,&#39;image_Aging4.sumstats.parquet&#39;,&#39;UKB.Lym.BOLT.sumstats.parquet&#39;,&#39;UKB.Mono.BOLT.sumstats.parquet&#39;,&#39;UKB.Plt.BOLT.sumstats.parquet&#39;,&#39;UKB.RBC.BOLT.sumstats.parquet&#39;,&#39;UKB_460K.blood_RBC_DISTRIB_WIDTH.sumstats.parquet&#39;,&#39;UKB.Eosino.BOLT.sumstats.parquet&#39;,&#39;UKB.Baso.BOLT.sumstats.parquet&#39;,&#39;UKB.Neutro.BOLT.sumstats.parquet&#39;,&#39;UKB.MCV.BOLT.sumstats.parquet&#39;,&#39;UKB_460K.biochemistry_Creatinine.sumstats.parquet&#39;,&#39;UKB.VitD.BOLT.sumstats.parquet&#39;,&#39;UKB.TBil.BOLT.sumstats.parquet&#39;,&#39;UKB_460K.biochemistry_AlkalinePhosphatase.sumstats.parquet&#39;,&#39;UKB_460K.biochemistry_AspartateAminotransferase.sumstats.parquet&#39;,&#39;UKB.TP.BOLT.sumstats.parquet&#39;,&#39;PASS_IBD_deLange2017.sumstats.parquet&#39;,&#39;PASS_CD_deLange2017.sumstats.parquet&#39;,&#39;PASS_UC_deLange2017.sumstats.parquet&#39;,&#39;UKB_460K.disease_ALLERGY_ECZEMA_DIAGNOSED.sumstats.parquet&#39;,&#39;UKB.Hypothyroidism.SAIGE.sumstats.parquet&#39;,&#39;PASS_Rheumatoid_Arthritis.sumstats.parquet&#39;,&#39;PASS_Primary_biliary_cirrhosis.sumstats.parquet&#39;,&#39;PASS_Lupus.sumstats.parquet&#39;,&#39;PASS_Type_1_Diabetes.sumstats.parquet&#39;,&#39;UKB.AID_Combined.SAIGE.sumstats.parquet&#39;,&#39;PASS_Celiac.sumstats.parquet&#39;,&#39;PASS_Alzheimers_Jansen2019.sumstats.parquet&#39;,&#39;PASS_Multiple_sclerosis.sumstats.parquet&#39;,&#39;UKB_460K.repro_NumberChildrenEverBorn_Pooled.sumstats.parquet&#39;,&#39;PASS_Anorexia.sumstats.parquet&#39;,&#39;PASS_ADHD_Demontis2018.sumstats.parquet&#39;,&#39;PASS_Autism.sumstats.parquet&#39;,&#39;PASS_SleepDuration_Dashti2019.sumstats.parquet&#39;,&#39;UKB.BMI.BOLT.sumstats.parquet&#39;,&#39;PASS_MDD_Wray2018.sumstats.parquet&#39;,&#39;PASS_Neuroticism.sumstats.parquet&#39;,&#39;UKB.Smoking_Ever_Never.SAIGE.sumstats.parquet&#39;,&#39;UKB_460K.cov_EDU_YEARS.sumstats.parquet&#39;,&#39;PASS_Intelligence_SavageJansen2018.sumstats.parquet&#39;,&#39;UKB.Morning_Person.BOLT.sumstats.parquet&#39;,&#39;UKB.Insomnia.BOLT.sumstats.parquet&#39;,&#39;PASS_Schizophrenia.sumstats.parquet&#39;,&#39;PASS_SCZvsBD_Ruderfer2018.sumstats.parquet&#39;,&#39;PASS_BipolarDisorder_Ruderfer2018.sumstats.parquet&#39;,&#39;PASS_ReactionTime_Davies2018.sumstats.parquet&#39;,&#39;PASS_AgeFirstBirth.sumstats.parquet&#39;,&#39;CAD_META.filtered.sumstats.parquet&#39;,&#39;UKB_460K.bp_SYSTOLICadjMEDz.sumstats.parquet&#39;,&#39;UKB_460K.bp_DIASTOLICadjMEDz.sumstats.parquet&#39;,&#39;PASS_ChildOnsetAsthma_Ferreira2019.sumstats.parquet&#39;,&#39;UKB.FEV1FVC.BOLT.sumstats.parquet&#39;,&#39;PASS_Height1.sumstats.parquet&#39;,&#39;UKB.BrC.SAIGE.sumstats.parquet&#39;,&#39;UKB.WHRadjBMI.BOLT.sumstats.parquet&#39;,&#39;Mahajan.NatGenet2018b.T2D.European.sumstats.parquet&#39;,&#39;UKB_460K.impedance_BASAL_METABOLIC_RATEz.sumstats.parquet&#39;,&#39;PASS_GeneralRiskTolerance_KarlssonLinner2019.sumstats.parquet&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(data.table)

traits_dt &lt;- rbindlist(lapply(seq_along(ttmp), function(i) {
  res &lt;- ttmp[[i]]
   enr &lt;- res$enrichment$enrichment_summary
  data.table(
    trait = names(ttmp)[i],
    enrichment = enr$Enrichment.Enrichment,
    se = enr$Enrichment_std_error.Enrichment_std_error,
    prop_h2 = enr$`Prop._h2.Prop._h2`,
    prop_snps = enr$`Prop._SNPs.Prop._SNPs`,
    p = enr$Enrichment_p,
    enrich_z = res$meta_enrstat$enrich_z,
    h2g = res$single_tau$h2g,
    sd_annot =as.numeric(res$single_tau$sd_annot)
  )
}))
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>traits_dt &lt;- traits_dt %&gt;% mutate(category = ifelse(trait %in% blood_traits, &#39;Blood&#39;, 
                                      ifelse(trait %in% brain_traits, &#39;Brain&#39;, 
                                      ifelse(trait %in% all_traits, &#39;Others&#39;, NA))))
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>traits_dt$trait &lt;- gsub(&#39;.sumstats.parquet&#39;, &#39;&#39;, traits_dt$trait)
traits_dt$trait &lt;- gsub(&#39;.munged.parquet&#39;, &#39;&#39;, traits_dt$trait)
traits_dt$trait &lt;- gsub(&#39;_munged.parquet&#39;, &#39;&#39;, traits_dt$trait)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>trait_names &lt;- c(

  # Alzheimer’s/ Dementia / FTD &amp; neurological disorders
  &quot;PASS_Alzheimers_Jansen2019&quot; = &quot;Alzheimer&#39;s Disease (Jansen 2019)&quot;,
  &quot;PASS_Multiple_sclerosis&quot;    = &quot;Multiple Sclerosis&quot;,
  &quot;AD_Bellenguez_buildGRCh38&quot;                    = &quot;Alzheimer&#39;s Disease (Bellenguez)&quot;,
  &quot;AD_Wightman_Excluding23andMe_hg38_sorted&quot;     = &quot;Alzheimer&#39;s Disease (Wightman excl. 23andMe)&quot;,
  &quot;AD_Bellenguez_EADB_hg38_sorted&quot;               = &quot;Alzheimer&#39;s Disease (Bellenguez EADB)&quot;,
  &quot;AD_Wightman_ExcludingUKB23andMe_hg38_sorted&quot;  = &quot;Alzheimer&#39;s Disease (Wightman excl. UKB &amp; 23andMe)&quot;,
  &quot;AD_Kunkle_etal_Stage1_results_hg38.sorted&quot;    = &quot;Alzheimer&#39;s Disease (Kunkle Stage 1)&quot;,
  &quot;AD_Wightman_full_2021&quot;                        = &quot;Alzheimer&#39;s Disease (Wightman Full)&quot;,

  # Brain structures / imaging / neuro
  &quot;EXF&quot;                        = &quot;Executive Function&quot;,
  &quot;LAN&quot;                        = &quot;Language&quot;,
  &quot;MEM&quot;                        = &quot;Memory&quot;,
  &quot;MS&quot;                         = &quot;Multiple Sclerosis (from Lu)&quot;,
  &quot;accumbens&quot;                  = &quot;Nucleus Accumbens Volume&quot;,
  &quot;ALS-Rheenen-2021.GCST90027163_buildGRCh37&quot; = &quot;ALS (Rheenen 2021, GWAS1)&quot;,
  &quot;ALS-Rheenen-2021.GCST90027164_buildGRCh37&quot; = &quot;ALS (Rheenen 2021, GWAS2)&quot;,
  &quot;amygdala&quot;                   = &quot;Amygdala Volume&quot;,
  &quot;brainstem&quot;                  = &quot;Brainstem Volume&quot;,
  &quot;caudate&quot;                    = &quot;Caudate Volume&quot;,
  &quot;GWAS_CP_all&quot;                = &quot;Cognitive Performance&quot;,
  &quot;GWAS_EA_excl23andMe&quot;        = &quot;Educational Attainment (Excl. 23andMe)&quot;,
  &quot;MegaGWAS_summary_European&quot;  = &quot;Cognitive Ability (European MegaGWAS)&quot;,
  &quot;nallsEtAl2019_excluding23andMe&quot; = &quot;Parkinson&#39;s Disease (Nalls 2019 excl. 23andMe)&quot;,
  &quot;neuroticism_ctg&quot;            = &quot;Neuroticism (CTG)&quot;,
  &quot;pallidum&quot;                   = &quot;Pallidum Volume&quot;,
  &quot;PGC3_SCZ_wave3_public.v2&quot;   = &quot;Schizophrenia (PGC3 Wave3)&quot;,
  &quot;PGC_UKB_23andMe_depression_10000&quot; = &quot;Depression (PGC+UKB+23andMe)&quot;,
  &quot;PGC_UKB_depression_genome-wide&quot;   = &quot;Depression (PGC+UKB GWAS)&quot;,
  &quot;putamen&quot;                    = &quot;Putamen Volume&quot;,
  &quot;rmf.surf&quot;                   = &quot;Rostral Middle Frontal Surface Area&quot;,
  &quot;rmf.thick&quot;                  = &quot;Rostral Middle Frontal Thickness&quot;,
  &quot;sf.surf&quot;                    = &quot;Superior Frontal Surface Area&quot;,
  &quot;sf.thick&quot;                   = &quot;Superior Frontal Thickness&quot;,
  &quot;image_AD1&quot;                  = &quot;Imaging Alzheimer&#39;s Set1&quot;,
  &quot;image_AD2&quot;                  = &quot;Imaging Alzheimer&#39;s Set2&quot;,
  &quot;image_Aging1&quot;               = &quot;Imaging Aging Set1&quot;,
  &quot;image_Aging2&quot;               = &quot;Imaging Aging Set2&quot;,
  &quot;image_Aging3&quot;               = &quot;Imaging Aging Set3&quot;,
  &quot;image_Aging4&quot;               = &quot;Imaging Aging Set4&quot;,
  &quot;image_Aging5&quot;               = &quot;Imaging Aging Set5&quot;,

  # Blood-related traits
  &quot;UKB.Lym.BOLT&quot;               = &quot;UKB Lymphocyte Count&quot;,
  &quot;UKB.Mono.BOLT&quot;              = &quot;UKB Monocyte Count&quot;,
  &quot;UKB.Plt.BOLT&quot;               = &quot;UKB Platelet Count&quot;,
  &quot;UKB.RBC.BOLT&quot;               = &quot;UKB RBC Count&quot;,
  &quot;UKB_460K.blood_RBC_DISTRIB_WIDTH&quot; = &quot;UKB RBC Distribution Width&quot;,
  &quot;UKB.Eosino.BOLT&quot;            = &quot;UKB Eosinophil Count&quot;,
  &quot;UKB.Baso.BOLT&quot;              = &quot;UKB Basophil Count&quot;,
  &quot;UKB.Neutro.BOLT&quot;            = &quot;UKB Neutrophil Count&quot;,
  &quot;UKB.MCV.BOLT&quot;               = &quot;UKB Mean Corpuscular Volume&quot;,
  &quot;UKB_460K.biochemistry_Creatinine&quot; = &quot;UKB Serum Creatinine&quot;,
  &quot;UKB.VitD.BOLT&quot;              = &quot;UKB Vitamin D&quot;,
  &quot;UKB.TBil.BOLT&quot;              = &quot;UKB Total Bilirubin&quot;,
  &quot;UKB_460K.biochemistry_AlkalinePhosphatase&quot; = &quot;UKB Alkaline Phosphatase&quot;,
  &quot;UKB_460K.biochemistry_AspartateAminotransferase&quot; = &quot;UKB AST&quot;,
  &quot;UKB.TP.BOLT&quot;                = &quot;UKB Total Protein&quot;,

  # Autoimmune &amp; Inflammatory
  &quot;PASS_IBD_deLange2017&quot;       = &quot;Inflammatory Bowel Disease (de Lange 2017)&quot;,
  &quot;PASS_CD_deLange2017&quot;        = &quot;Crohn&#39;s Disease (de Lange 2017)&quot;,
  &quot;PASS_UC_deLange2017&quot;        = &quot;Ulcerative Colitis (de Lange 2017)&quot;,
  &quot;UKB_460K.disease_ALLERGY_ECZEMA_DIAGNOSED&quot; = &quot;Allergy/Eczema (UKB)&quot;,
  &quot;UKB.Hypothyroidism.SAIGE&quot;   = &quot;Hypothyroidism (UKB)&quot;,
  &quot;PASS_Rheumatoid_Arthritis&quot;  = &quot;Rheumatoid Arthritis&quot;,
  &quot;PASS_Primary_biliary_cirrhosis&quot; = &quot;Primary Biliary Cirrhosis&quot;,
  &quot;PASS_Lupus&quot;                 = &quot;Systemic Lupus Erythematosus&quot;,
  &quot;PASS_Type_1_Diabetes&quot;       = &quot;Type 1 Diabetes&quot;,
  &quot;UKB.AID_Combined.SAIGE&quot;     = &quot;Combined Autoimmune Diseases (UKB)&quot;,
  &quot;PASS_Celiac&quot;                = &quot;Celiac Disease&quot;,


  # Reproductive &amp; psychiatric
  &quot;UKB_460K.repro_NumberChildrenEverBorn_Pooled&quot; = &quot;Number of Children Ever Born&quot;,
  &quot;PASS_Anorexia&quot;              = &quot;Anorexia Nervosa&quot;,
  &quot;PASS_ADHD_Demontis2018&quot;     = &quot;ADHD (Demontis 2018)&quot;,
  &quot;PASS_Autism&quot;                = &quot;Autism Spectrum Disorder&quot;,
  &quot;PASS_SleepDuration_Dashti2019&quot; = &quot;Sleep Duration (Dashti 2019)&quot;,
  &quot;UKB.BMI.BOLT&quot;               = &quot;UKB BMI&quot;,
  &quot;PASS_MDD_Wray2018&quot;          = &quot;Major Depression (Wray 2018)&quot;,
  &quot;PASS_Neuroticism&quot;           = &quot;Neuroticism (PASS)&quot;,
  &quot;UKB.Smoking_Ever_Never.SAIGE&quot; = &quot;Smoking Ever/Never (UKB)&quot;,
  &quot;UKB_460K.cov_EDU_YEARS&quot;     = &quot;Educational Years (UKB)&quot;,
  &quot;PASS_Intelligence_SavageJansen2018&quot; = &quot;Intelligence (Savage &amp; Jansen 2018)&quot;,
  &quot;UKB.Morning_Person.BOLT&quot;    = &quot;Chronotype (Morning Person)&quot;,
  &quot;UKB.Insomnia.BOLT&quot;          = &quot;Insomnia (UKB)&quot;,
  &quot;PASS_Schizophrenia&quot;         = &quot;Schizophrenia&quot;,
  &quot;PASS_SCZvsBD_Ruderfer2018&quot;  = &quot;Schizophrenia vs Bipolar (Ruderfer 2018)&quot;,
  &quot;PASS_BipolarDisorder_Ruderfer2018&quot; = &quot;Bipolar Disorder (Ruderfer 2018)&quot;,
  &quot;PASS_ReactionTime_Davies2018&quot; = &quot;Reaction Time (Davies 2018)&quot;,
  &quot;PASS_AgeFirstBirth&quot;         = &quot;Age at First Birth&quot;,

  # Cardiometabolic &amp; general health
  &quot;CAD_META.filtered&quot;          = &quot;Coronary Artery Disease (Meta)&quot;,
  &quot;UKB_460K.bp_SYSTOLICadjMEDz&quot; = &quot;Systolic BP (adj. meds)&quot;,
  &quot;UKB_460K.bp_DIASTOLICadjMEDz&quot; = &quot;Diastolic BP (adj. meds)&quot;,
  &quot;PASS_ChildOnsetAsthma_Ferreira2019&quot; = &quot;Childhood Onset Asthma (Ferreira 2019)&quot;,
  &quot;UKB.FEV1FVC.BOLT&quot;           = &quot;FEV1/FVC Ratio (Lung Function)&quot;,
  &quot;PASS_Height1&quot;               = &quot;Height&quot;,
  &quot;UKB.BrC.SAIGE&quot;              = &quot;Breast Cancer (UKB)&quot;,
  &quot;UKB.WHRadjBMI.BOLT&quot;         = &quot;Waist-Hip Ratio adj. BMI&quot;,
  &quot;Mahajan.NatGenet2018b.T2D.European&quot; = &quot;Type 2 Diabetes (Mahajan 2018, European)&quot;,
  &quot;UKB_460K.impedance_BASAL_METABOLIC_RATEz&quot; = &quot;Basal Metabolic Rate (UKB)&quot;,
  &quot;PASS_GeneralRiskTolerance_KarlssonLinner2019&quot; = &quot;General Risk Tolerance (Karlsson Linner 2019)&quot;
)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(dplyr)

# replace codes with human-readable names
traits_dt &lt;- traits_dt %&gt;%
  mutate(trait = recode(trait, !!!trait_names))
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>traits_dt$category &lt;- factor(traits_dt$category, levels = c(&#39;Brain&#39;, &#39;Blood&#39;, &#39;Others&#39;))
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>options(repr.plot.width = 25, repr.plot.height = 10)

library(ggplot2)
library(RColorBrewer)

category_colors &lt;- c(
  &quot;Blood&quot; = &quot;#4575B4&quot;,     # 
  &quot;Brain&quot; = &quot;#D73027&quot;,     # 
  &quot;All&quot; = &quot;#1A9850&quot;,    # 
  &quot;Others&quot; = &quot;#984EA3&quot;      # others
)

# Assign colors by category (soft variations within each)
traits_dt &lt;- traits_dt %&gt;%
  group_by(category) %&gt;%
  mutate(
    trait_order = as.integer(factor(trait, levels = unique(trait[order(enrichment)]))),
    n_traits = n(),
    color = lighten(category_colors[category], 
                    amount = scales::rescale(trait_order, to = c(0.6, 0.1)))
  ) %&gt;%
  ungroup()
ggplot(traits_dt, aes(x = reorder(trait, enrichment), y = enrichment, color = trait)) +
  geom_point(size = 3) +
  geom_errorbar(aes(ymin = enrichment - se, ymax = enrichment + se), width = 0.3) +
  coord_flip() +
  facet_wrap(~category, scales = &#39;free_y&#39;) +
  labs(
    title = &quot;LDSC Enrichment Across Traits&quot;,
    x = &quot;Trait&quot;,
    y = &quot;Enrichment ± SE&quot;
  ) +
  scale_color_manual(values = setNames(traits_dt$color, traits_dt$trait)) +
  theme_minimal(base_size = 16) +
  theme(
    strip.text = element_text(face = &quot;bold&quot;, size = 18),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    axis.title = element_text(size = 14),
    legend.position = &quot;none&quot;,
    panel.grid.minor = element_blank(),
    panel.grid.major.y = element_blank()
  )
</pre></div>
</div>
</div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "sos"
        },
        kernelOptions: {
            name: "sos",
            path: "./main_text/2_single_context_cis/staging"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'sos'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#plot">Plot</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#meta-traits">Meta traits</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#seprate-traits">Seprate traits</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>