
<!DOCTYPE html>


<html lang="en" data-content_root="../../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Figure 2c. Marginal vs multicontext effect sign heatmap &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/4_multi_context/staging/Figure 2c';</script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../1_project_overview/index.html">FunGen-xQTL atlas and companion projects</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../2_single_context_cis/index.html">Single context cis-xQTL fine-mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../3_single_context_multigene_trans/index.html">Multi-gene and trans-xQTL fine-mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../index.html">Propagation of multi-context xQTL effects</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../5_AD_xQTL_integration/index.html">Alzheimer’s disease (AD) loci integration</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5a_AD_GWAS_loci_harmonized.html">AD GWAS loci harmonization and refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5b_AD_GWAS_loci_properties.html">Refined AD GWAS loci properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5c_AD_loci_xQTL_annotated.html">AD loci “explained” by xQTL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5d_AD_loci_table_summary.html">Summary of FunGen-xQTL AD loci table</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5e_GSEA.html">Gene Set Enrichment Analysis for AD Risk Genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5f_PPI.html">PPI Networks for Selected AD Risk Genes in xQTL Loci</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../5_AD_xQTL_integration/Figure_5g_cV2F.html">Consensus Variant-to-Function Score for AD</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../6_AD_xQTL_genes/index.html">Insights into AD risk genes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.html">Visualization of FunGen-xQTL AD loci table</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/4_multi_context/staging/Figure 2c.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../../_sources/main_text/4_multi_context/staging/Figure 2c.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Figure 2c. Marginal vs multicontext effect sign heatmap</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#marginal-effect-sign-heatmap">Marginal effect sign heatmap</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#posterior-table">posterior table</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#trait-matrix">trait_matrix</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#heatmap">heatmap</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multicontext-effect-sign-heatmap">multicontext effect sign heatmap</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">trait_matrix</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">heatmap</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="figure-2c-marginal-vs-multicontext-effect-sign-heatmap">
<h1>Figure 2c. Marginal vs multicontext effect sign heatmap<a class="headerlink" href="#figure-2c-marginal-vs-multicontext-effect-sign-heatmap" title="Link to this heading">#</a></h1>
<section id="marginal-effect-sign-heatmap">
<h2>Marginal effect sign heatmap<a class="headerlink" href="#marginal-effect-sign-heatmap" title="Link to this heading">#</a></h2>
<section id="posterior-table">
<h3>posterior table<a class="headerlink" href="#posterior-table" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">reshape2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="c1"># replace with the new posterior path in the folder &quot;interactive_sessions/aw3600/mash_prior_0718_posterior/mash_34_contexts_ed_bovy_1e3.EZ.posterior.rds&quot;</span>
<span class="n">posterior_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;~/project/Figure_2b/adgwas_20k_prune37_ed_bovy_1e3.EZ.posterior.rds&quot;</span><span class="p">)</span>
<span class="c1">### transfer the lfsr to the four columns</span>
<span class="n">posterior_dat_lfsr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">lfsr</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">                             </span><span class="n">split_info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">lfsr</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span>
<span class="w">                             </span><span class="n">variant</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span>
<span class="w">                             </span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="p">]</span>
<span class="w">                           </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">split_info</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">pivot_longer</span><span class="p">(</span>
<span class="w">                            </span><span class="n">cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="n">variant</span><span class="p">,</span><span class="w"> </span><span class="n">gene</span><span class="p">),</span><span class="w">  </span><span class="c1"># Keep &#39;variant&#39; and &#39;gene&#39; columns fixed</span>
<span class="w">                            </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;context&quot;</span><span class="p">,</span><span class="w">      </span><span class="c1"># New column for context (column names)</span>
<span class="w">                            </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;lfsr&quot;</span><span class="w">  </span><span class="c1"># New column for effect sizes (values)</span>
<span class="w">                     </span><span class="p">)</span><span class="w"> </span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat_lfsr</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span>
<span class="n">posterior_dat_negative_prob</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">NegativeProb</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">                             </span><span class="n">split_info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">NegativeProb</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span>
<span class="w">                             </span><span class="n">variant</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span>
<span class="w">                             </span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="p">]</span>
<span class="w">                           </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">split_info</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">pivot_longer</span><span class="p">(</span>
<span class="w">                            </span><span class="n">cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="n">variant</span><span class="p">,</span><span class="w"> </span><span class="n">gene</span><span class="p">),</span><span class="w">  </span><span class="c1"># Keep &#39;variant&#39; and &#39;gene&#39; columns fixed</span>
<span class="w">                            </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;context&quot;</span><span class="p">,</span><span class="w">      </span><span class="c1"># New column for context (column names)</span>
<span class="w">                            </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;prob&quot;</span><span class="w">  </span><span class="c1"># New column for effect sizes (values)</span>
<span class="w">                     </span><span class="p">)</span>
<span class="c1">### transfer the posterior_mean to the four columns</span>
<span class="n">posterior_dat_effect</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">PosteriorMean</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">                             </span><span class="n">split_info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">PosteriorMean</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span>
<span class="w">                             </span><span class="n">variant</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span>
<span class="w">                             </span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="p">]</span>
<span class="w">                           </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">split_info</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">pivot_longer</span><span class="p">(</span>
<span class="w">                            </span><span class="n">cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="n">variant</span><span class="p">,</span><span class="w"> </span><span class="n">gene</span><span class="p">),</span><span class="w">  </span><span class="c1"># Keep &#39;variant&#39; and &#39;gene&#39; columns fixed</span>
<span class="w">                            </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;context&quot;</span><span class="p">,</span><span class="w">      </span><span class="c1"># New column for context (column names)</span>
<span class="w">                            </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;effect&quot;</span><span class="w">  </span><span class="c1"># New column for effect sizes (values)</span>
<span class="w">                     </span><span class="p">)</span>
<span class="n">posterior_dat_effect</span><span class="o">$</span><span class="n">effect_sign</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">posterior_dat_effect</span><span class="o">$</span><span class="n">effect</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span>
<span class="w">                                           </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span>
<span class="w">                                           </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;+1&quot;</span><span class="p">))</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat_effect</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span>
<span class="c1">### integrate lfsr and effec, also effect sign</span>
<span class="n">posterior_dat_lfsr_effect</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr</span><span class="o">%&gt;%</span>
<span class="w">                                </span><span class="nf">left_join</span><span class="p">(</span><span class="n">posterior_dat_effect</span><span class="p">,</span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;variant&quot;</span><span class="p">,</span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="s">&quot;context&quot;</span><span class="p">))</span><span class="o">%&gt;%</span>
<span class="w">                                </span><span class="nf">left_join</span><span class="p">(</span><span class="n">posterior_dat_negative_prob</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;variant&quot;</span><span class="p">,</span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="s">&quot;context&quot;</span><span class="p">))</span>
<span class="n">lfsr_cutoff</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0.01</span>
<span class="n">posterior_dat_lfsr_effect_filtered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">lfsr</span><span class="o">&lt;</span><span class="n">lfsr_cutoff</span><span class="p">)</span>
<span class="n">posterior_dat_lfsr_effect_filtered_context_top_variants</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                                                                </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">context</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                                                                </span><span class="nf">filter</span><span class="p">(</span><span class="n">lfsr</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">min</span><span class="p">(</span><span class="n">lfsr</span><span class="p">),</span><span class="n">prob</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">prob</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                                                </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                                                </span><span class="nf">unique</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">intersected_gene_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_top_variants</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">context</span><span class="o">%in%</span><span class="n">traits</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">%in%</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                                                           </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">))))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">unique</span><span class="p">()</span>
<span class="nf">length</span><span class="p">(</span><span class="n">intersected_gene_names</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">4037</div></div>
</div>
</section>
<section id="trait-matrix">
<h3>trait_matrix<a class="headerlink" href="#trait-matrix" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#trait_matrix &lt;- readRDS(&quot;~/project/Figure_2b/marginal_efect_sign_0429_4063.rds&quot;)</span>
<span class="c1"># replace with the new version multicontext top loci table</span>
<span class="n">ROSMAP_multicontext_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;/data/analysis_result/multi_context/ROSMAP/export/summary/ROSMAP.exported.toploci.bed.gz&quot;</span><span class="p">)</span>
<span class="c1"># we split the event_ID in the multicontext top loci table to get `mvsusie_export_lfsr_integrated_df`</span>
<span class="n">mvsusie_export_lfsr_integrated_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;~/project/Figure_2b/multicontext_top_loci_integrated.rds&quot;</span><span class="p">)</span>
<span class="n">traits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">event_ID</span><span class="p">)))</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">intersected_gene_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_top_variants</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">context</span><span class="o">%in%</span><span class="n">traits</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">%in%</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                                                           </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">))))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">unique</span><span class="p">()</span>
<span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_res</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="o">%in%</span><span class="n">intersected_gene_names</span><span class="p">)</span>
<span class="n">posterior_dat_lfsr_effect_filtered_context</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">context</span><span class="o">%in%</span><span class="n">traits</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">%in%</span><span class="nf">unique</span><span class="p">(</span><span class="n">ROSMAP_multicontext_extracted</span><span class="o">$</span><span class="n">gene_ID</span><span class="p">))</span>
<span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">variant</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">summarise</span><span class="p">(</span>
<span class="w">    </span><span class="n">contexts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">context</span><span class="p">),</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">),</span><span class="w"> </span>
<span class="w">    </span><span class="n">.groups</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;drop&quot;</span>
<span class="w">  </span><span class="p">)</span>
<span class="n">num_traits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">traits</span><span class="p">)</span>
<span class="n">trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">num_traits</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">num_traits</span><span class="p">)</span><span class="w">  </span><span class="c1"># Initialize matrix</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">traits</span><span class="w">   </span><span class="c1"># Assign trait names</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">gene_name</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="o">$</span><span class="n">gene</span><span class="p">)){</span>
<span class="c1">#gene_name &lt;- as.character(unique(mvsusie_export_lfsr_integrated_df$gene))[1]</span>
<span class="n">strong_variants_lfsr_thres</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">gene_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">unique</span><span class="p">()</span>
<span class="c1">#Iterate through variants in df2</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">strong_variants_lfsr_thres</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1"># Extract rows from df1 where the marker matches the variant in df2</span>
<span class="w">  </span><span class="n">variant_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">==</span><span class="n">gene_name</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">variant</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">variant_ID</span><span class="p">)</span>
<span class="w">  </span><span class="c1"># If variant is found in multiple traits, compare effect signs</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span>
<span class="w">    </span><span class="c1"># Get unique trait pairs</span>
<span class="w">    </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span>
<span class="w">        </span><span class="n">trait_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">context</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
<span class="w">        </span><span class="n">trait_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">context</span><span class="p">[</span><span class="n">j</span><span class="p">]</span>
<span class="w">        </span>
<span class="w">        </span><span class="n">effect_sign_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">effect_sign</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
<span class="w">        </span><span class="n">effect_sign_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">effect_sign</span><span class="p">[</span><span class="n">j</span><span class="p">]</span>

<span class="w">        </span><span class="c1"># Find indices in the trait matrix</span>
<span class="w">        </span><span class="n">idx_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">trait_i</span><span class="p">)</span>
<span class="w">        </span><span class="n">idx_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">trait_j</span><span class="p">)</span>
<span class="w">        </span>
<span class="w">        </span><span class="c1"># Ensure idx_i is always smaller than idx_j for upper triangle updates</span>
<span class="w">        </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">idx_i</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">idx_j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">              </span><span class="n">temp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">idx_i</span>
<span class="w">              </span><span class="n">idx_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">idx_j</span>
<span class="w">              </span><span class="n">idx_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">temp</span>
<span class="w">        </span><span class="p">}</span>

<span class="w">        </span><span class="c1"># Update matrix correctly with only +1 increments</span>
<span class="w">        </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">effect_sign_i</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">effect_sign_j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">            </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_i</span><span class="p">,</span><span class="w"> </span><span class="n">idx_j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_i</span><span class="p">,</span><span class="w"> </span><span class="n">idx_j</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Always upper triangle</span>
<span class="w">        </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">            </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_j</span><span class="p">,</span><span class="w"> </span><span class="n">idx_i</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_j</span><span class="p">,</span><span class="w"> </span><span class="n">idx_i</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Always lower triangle</span>
<span class="w">        </span><span class="p">}</span>
<span class="w">      </span><span class="p">}</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">trait_idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">context</span><span class="p">)</span>
<span class="w">    </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">trait_idx</span><span class="p">,</span><span class="w"> </span><span class="n">trait_idx</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">trait_idx</span><span class="p">,</span><span class="w"> </span><span class="n">trait_idx</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="c1"># Diagonal increment</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>

<span class="n">trait_matrix</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A matrix: 11 × 11 of type dbl</caption>
<thead>
	<tr><th></th><th scope=col>AC_DeJager_eQTL</th><th scope=col>DLPFC_DeJager_eQTL</th><th scope=col>PCC_DeJager_eQTL</th><th scope=col>Exc_DeJager_eQTL</th><th scope=col>Inh_DeJager_eQTL</th><th scope=col>Oli_DeJager_eQTL</th><th scope=col>Ast_DeJager_eQTL</th><th scope=col>DLPFC_Bennett_pQTL</th><th scope=col>monocyte_ROSMAP_eQTL</th><th scope=col>OPC_DeJager_eQTL</th><th scope=col>Mic_DeJager_eQTL</th></tr>
</thead>
<tbody>
	<tr><th scope=row>AC_DeJager_eQTL</th><td>188</td><td>3288</td><td>3511</td><td>2080</td><td>1713</td><td>1616</td><td>1793</td><td> 885</td><td>398</td><td>1095</td><td>675</td></tr>
	<tr><th scope=row>DLPFC_DeJager_eQTL</th><td> 86</td><td>  79</td><td>3889</td><td>2791</td><td>2054</td><td>1765</td><td>1872</td><td> 969</td><td>391</td><td>1240</td><td>658</td></tr>
	<tr><th scope=row>PCC_DeJager_eQTL</th><td> 33</td><td>  24</td><td>  18</td><td>2657</td><td>1961</td><td>1588</td><td>1799</td><td>1038</td><td>384</td><td>1147</td><td>652</td></tr>
	<tr><th scope=row>Exc_DeJager_eQTL</th><td>504</td><td> 247</td><td> 309</td><td> 162</td><td>2975</td><td>1715</td><td>1770</td><td> 715</td><td>337</td><td>1345</td><td>796</td></tr>
	<tr><th scope=row>Inh_DeJager_eQTL</th><td>431</td><td> 277</td><td> 328</td><td>  64</td><td>  92</td><td>1626</td><td>1717</td><td> 466</td><td>303</td><td>1347</td><td>798</td></tr>
	<tr><th scope=row>Oli_DeJager_eQTL</th><td>308</td><td> 217</td><td> 291</td><td> 285</td><td> 216</td><td> 141</td><td>1594</td><td> 384</td><td>296</td><td>1320</td><td>728</td></tr>
	<tr><th scope=row>Ast_DeJager_eQTL</th><td>233</td><td> 177</td><td> 210</td><td> 248</td><td> 167</td><td> 168</td><td>  63</td><td> 437</td><td>270</td><td>1388</td><td>761</td></tr>
	<tr><th scope=row>DLPFC_Bennett_pQTL</th><td> 96</td><td>  86</td><td>  64</td><td> 200</td><td> 230</td><td> 186</td><td> 148</td><td> 200</td><td>116</td><td> 241</td><td>166</td></tr>
	<tr><th scope=row>monocyte_ROSMAP_eQTL</th><td> 78</td><td>  70</td><td>  84</td><td> 106</td><td>  87</td><td>  62</td><td>  52</td><td>  55</td><td>127</td><td> 189</td><td>271</td></tr>
	<tr><th scope=row>OPC_DeJager_eQTL</th><td>227</td><td> 151</td><td> 183</td><td> 172</td><td>  99</td><td> 114</td><td>  81</td><td> 138</td><td> 37</td><td>  85</td><td>641</td></tr>
	<tr><th scope=row>Mic_DeJager_eQTL</th><td>166</td><td> 148</td><td> 160</td><td> 142</td><td> 116</td><td> 139</td><td>  98</td><td> 107</td><td> 21</td><td>  68</td><td> 63</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="heatmap">
<h3>heatmap<a class="headerlink" href="#heatmap" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Ensure &quot;AC_DeJager_eQTL&quot; stays on top</span>
<span class="c1">#trait_matrix &lt;- readRDS(&quot;~/project/Figure_2b/marginal_efect_sign_0429_4063.rds&quot;)</span>
<span class="n">trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))),</span><span class="w"> </span>
<span class="w">                           </span><span class="nf">order</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)))]</span>
<span class="n">rev_trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[,</span><span class="nf">ncol</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="c1">#rev_trait_matrix &lt;- t(trait_matrix)</span>
<span class="n">rev_trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_trait_matrix</span><span class="p">)</span>

<span class="c1"># Calculate the fraction matrix</span>
<span class="c1"># Assuming rev_trait_matrix is your input square matrix</span>
<span class="n">context_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">rev_trait_matrix</span><span class="p">)</span>
<span class="n">n</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">context_names</span><span class="p">)</span>

<span class="c1"># Initialize output matrix</span>
<span class="n">proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">context_names</span><span class="w"> </span>

<span class="n">pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">context_names</span><span class="w"> </span>

<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">seq_len</span><span class="p">(</span><span class="n">n</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">seq_len</span><span class="p">(</span><span class="n">n</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="n">j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="c1"># Extract 2x2 submatrix</span>
<span class="w">      </span><span class="n">trait_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)[</span><span class="n">i</span><span class="p">]</span>
<span class="w">      </span><span class="n">trait_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)[</span><span class="n">j</span><span class="p">]</span>
<span class="w">      </span><span class="n">pattern</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_c</span><span class="p">(</span><span class="n">trait_i</span><span class="p">,</span><span class="w"> </span><span class="n">trait_j</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;|&quot;</span><span class="p">)</span>
<span class="w">      </span><span class="n">submat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">)]</span>
<span class="w">      </span><span class="n">marginal_multicontext_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w">  </span><span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">contexts</span><span class="p">,</span><span class="n">pattern</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                                        </span><span class="nf">unique</span><span class="p">()</span><span class="w"> </span>
<span class="w">      </span><span class="n">pairwise_total_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">submat</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="nf">diag</span><span class="p">(</span><span class="n">submat</span><span class="p">))</span>
<span class="w">      </span><span class="n">total_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise_total_sum</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="nf">nrow</span><span class="p">(</span><span class="n">marginal_multicontext_res</span><span class="p">)</span>
<span class="w">      </span><span class="n">value</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span>

<span class="w">      </span><span class="c1"># Compute and assign proportion</span>
<span class="w">      </span><span class="n">proportion_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">round</span><span class="p">((</span><span class="n">value</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_sum</span><span class="p">)</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
<span class="w">      </span><span class="n">pairwise_proportion_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">round</span><span class="p">((</span><span class="n">value</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">pairwise_total_sum</span><span class="p">)</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>
<span class="c1"># View or use proportion_matrix</span>
<span class="c1">#rev_proportion_matrix</span>

<span class="n">rev_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">proportion_matrix</span><span class="p">[,</span><span class="w"> </span><span class="nf">ncol</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="n">rev_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)</span>


<span class="c1"># Convert matrix into long format for ggplot</span>
<span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">reshape2</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Trait1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Trait2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Count&quot;</span><span class="p">)</span>

<span class="c1"># 1. Identify which rows need filling</span>
<span class="n">na_idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">is.na</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Count</span><span class="p">)</span>

<span class="c1"># 2. Pull from the matrix by name for those rows</span>
<span class="c1"># Find numeric row/col indices</span>
<span class="n">r</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">match</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Trait1</span><span class="p">[</span><span class="n">na_idx</span><span class="p">],</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))</span>
<span class="n">c</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">match</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Trait2</span><span class="p">[</span><span class="n">na_idx</span><span class="p">],</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))</span>

<span class="c1"># Do element-wise lookup</span>
<span class="n">df</span><span class="o">$</span><span class="n">Count</span><span class="p">[</span><span class="n">na_idx</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">cbind</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="w"> </span><span class="n">c</span><span class="p">)]</span>

<span class="c1">#proportion_matrix</span>
<span class="n">rev_pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise_proportion_matrix</span><span class="p">[,</span><span class="w"> </span><span class="nf">ncol</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="n">rev_pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_pairwise_proportion_matrix</span><span class="p">)</span>

<span class="w"> </span>
<span class="n">df</span><span class="o">$</span><span class="n">Fraction</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.vector</span><span class="p">(</span><span class="n">reshape2</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">rev_pairwise_proportion_matrix</span><span class="p">)</span><span class="o">$</span><span class="n">value</span><span class="p">)</span>

<span class="c1">#credible_sum &lt;- posterior_dat_lfsr_effect_filtered_context_grouped %&gt;% unique() %&gt;%nrow()</span>


<span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">df</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">rowwise</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">Label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Fraction</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">paste0</span><span class="p">(</span><span class="n">Fraction</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n(&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">Count</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;)&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">total_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">contexts</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">Trait1</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">nrow</span><span class="p">()</span>
<span class="w">      </span>
<span class="w">      </span><span class="nf">round</span><span class="p">((</span><span class="n">Count</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_genes</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">as.character</span><span class="p">()</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">ungroup</span><span class="p">()</span>
<span class="c1"># df$Label &lt;- ifelse(!is.na(df$Fraction), </span>
<span class="c1">#                    paste0(df$Fraction, &quot;\n(&quot;, df$Count, &quot;)&quot;), </span>
<span class="c1">#                    as.character(round((df$Count/credible_sum)*100,2)))</span>

<span class="n">df</span><span class="o">$</span><span class="n">Trait1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Trait1</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rev</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)))</span>
<span class="n">df</span><span class="o">$</span><span class="n">Trait2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Trait2</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">))</span>
<span class="c1"># df$Trait1 &lt;- factor(df$Trait1, levels = rev(rownames(rev_trait_matrix)))</span>
<span class="c1"># df$Trait2 &lt;- factor(df$Trait2, levels = rownames(rev_trait_matrix))</span>

<span class="c1">#pdf(&quot;ROSMAP_marginal_effect_sign_heatmap_version_0429_4063.pdf&quot;, width =10, height = 8)</span>
<span class="c1"># Plot heatmap using ggplot2</span>
<span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Trait1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Trait2</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Fraction</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_tile</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_fill_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">na.value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Label</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="c1"># Add text with count and fraction</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Proportion&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Marginal Posterior Effect Sign Heatmap&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">45</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">),</span>
<span class="w">        </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">),</span>
<span class="w">        </span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">25</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">        </span><span class="n">legend.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">       </span><span class="n">legend.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">))</span>
<span class="c1">#dev.off()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">14</span><span class="p">)</span>
<span class="n">p1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/4a7c1f699614045af9ab4bceb62beccdd0c44ef462e853d811c789b2475f70a3.png"><img alt="../../../_images/4a7c1f699614045af9ab4bceb62beccdd0c44ef462e853d811c789b2475f70a3.png" src="../../../_images/4a7c1f699614045af9ab4bceb62beccdd0c44ef462e853d811c789b2475f70a3.png" style="width: 1080px; height: 840px;" />
</a>
</div>
</div>
</section>
</section>
<section id="multicontext-effect-sign-heatmap">
<h2>multicontext effect sign heatmap<a class="headerlink" href="#multicontext-effect-sign-heatmap" title="Link to this heading">#</a></h2>
<section id="id1">
<h3>trait_matrix<a class="headerlink" href="#id1" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">mvsusie_export_lfsr_integrated_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;~/project/Figure_2b/multicontext_top_loci_integrated.rds&quot;</span><span class="p">)</span>
<span class="n">traits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">event_ID</span><span class="p">)))</span>
<span class="n">intersected_gene_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_top_variants</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">context</span><span class="o">%in%</span><span class="n">traits</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">%in%</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                                                           </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">))))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">unique</span><span class="p">()</span>
<span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_res</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="o">%in%</span><span class="n">intersected_gene_names</span><span class="p">)</span>
<span class="n">num_traits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">traits</span><span class="p">)</span>
<span class="n">trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">num_traits</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">num_traits</span><span class="p">)</span><span class="w">  </span><span class="c1"># Initialize matrix</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">traits</span><span class="w">   </span><span class="c1"># Assign trait names</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">gene_name</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">intersected_gene_names</span><span class="p">){</span>
<span class="c1">#gene_name &lt;- as.character(unique(mvsusie_export_lfsr_integrated_df$gene))[1]</span>
<span class="c1">#gene_name</span>
<span class="c1"># top_variants_lfsr_thres &lt;-  mvsusie_export_lfsr_integrated_df%&gt;%filter(gene_ID==gene_name)%&gt;%</span>
<span class="c1">#                             filter(.data[[coverage]]!=0)%&gt;%group_by(.data[[coverage]])%&gt;% </span>
<span class="c1">#                             filter(PIP == max(PIP, na.rm = TRUE)) %&gt;%</span>
<span class="c1">#                             distinct(variant_ID, .keep_all = TRUE) %&gt;%  # Ensure unique SNPs</span>
<span class="c1">#                             summarise(variant_ID = sample(variant_ID, 1), .groups = &quot;drop&quot;)%&gt;%</span>
<span class="c1">#                             pull(variant_ID) #top_variants_lfsr_thres</span>
<span class="n">top_variants_lfsr_thres</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_extracted</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="o">==</span><span class="n">gene_name</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                            </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">group_by</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]],</span><span class="n">event_ID</span><span class="p">)</span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                            </span><span class="nf">filter</span><span class="p">(</span><span class="n">PIP</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">PIP</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">slice_sample</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                            </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">unique</span><span class="p">()</span>
<span class="c1"># Step 2: Iterate through variants in df2</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">variant</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">top_variants_lfsr_thres</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1">#variant &lt;- &#39;chr20:50961558:T:C&#39;</span>
<span class="w">  </span><span class="c1"># Extract rows from df1 where the marker matches the variant in df2</span>
<span class="w">  </span><span class="n">variant_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="o">==</span><span class="n">gene_name</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">variant</span><span class="p">)</span>
<span class="w">  </span><span class="c1">#variant_data</span>
<span class="w">  </span><span class="c1"># If variant is found in multiple traits, compare effect signs</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span>
<span class="w">    </span><span class="c1"># Get unique trait pairs</span>
<span class="w">    </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span>
<span class="w">        </span><span class="n">trait_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">event_ID</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
<span class="w">        </span><span class="n">trait_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">event_ID</span><span class="p">[</span><span class="n">j</span><span class="p">]</span>
<span class="w">        </span>
<span class="w">        </span><span class="n">effect_sign_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">effect_sign</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
<span class="w">        </span><span class="n">effect_sign_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">effect_sign</span><span class="p">[</span><span class="n">j</span><span class="p">]</span>

<span class="w">        </span><span class="c1"># Find indices in the trait matrix</span>
<span class="n">idx_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">trait_i</span><span class="p">)</span>
<span class="n">idx_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">trait_j</span><span class="p">)</span>
<span class="w">        </span>
<span class="w">       </span><span class="c1"># Ensure idx_i is always smaller than idx_j for upper triangle updates</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">idx_i</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">idx_j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">temp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">idx_i</span>
<span class="w">  </span><span class="n">idx_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">idx_j</span>
<span class="w">  </span><span class="n">idx_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">temp</span>
<span class="p">}</span>

<span class="c1"># Update matrix correctly with only +1 increments</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">effect_sign_i</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">effect_sign_j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">          </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_i</span><span class="p">,</span><span class="w"> </span><span class="n">idx_j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_i</span><span class="p">,</span><span class="w"> </span><span class="n">idx_j</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Always upper triangle</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">          </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_j</span><span class="p">,</span><span class="w"> </span><span class="n">idx_i</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_j</span><span class="p">,</span><span class="w"> </span><span class="n">idx_i</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Always lower triangle</span>
<span class="p">}</span>
<span class="w">      </span><span class="p">}</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 3: Check if the variant is **context-specific**</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">trait_idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">event_ID</span><span class="p">)</span>
<span class="w">    </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">trait_idx</span><span class="p">,</span><span class="w"> </span><span class="n">trait_idx</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">trait_idx</span><span class="p">,</span><span class="w"> </span><span class="n">trait_idx</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Diagonal increment</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>
<span class="p">}</span>


<span class="n">trait_matrix</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A matrix: 11 × 11 of type dbl</caption>
<thead>
	<tr><th></th><th scope=col>AC_DeJager_eQTL</th><th scope=col>DLPFC_DeJager_eQTL</th><th scope=col>PCC_DeJager_eQTL</th><th scope=col>Exc_DeJager_eQTL</th><th scope=col>Inh_DeJager_eQTL</th><th scope=col>Oli_DeJager_eQTL</th><th scope=col>Ast_DeJager_eQTL</th><th scope=col>DLPFC_Bennett_pQTL</th><th scope=col>monocyte_ROSMAP_eQTL</th><th scope=col>OPC_DeJager_eQTL</th><th scope=col>Mic_DeJager_eQTL</th></tr>
</thead>
<tbody>
	<tr><th scope=row>AC_DeJager_eQTL</th><td>245</td><td>4945</td><td>4955</td><td>2877</td><td>2064</td><td>1833</td><td>2197</td><td>1276</td><td>527</td><td>1062</td><td>558</td></tr>
	<tr><th scope=row>DLPFC_DeJager_eQTL</th><td>217</td><td> 106</td><td>5264</td><td>3629</td><td>2456</td><td>2026</td><td>2400</td><td>1326</td><td>511</td><td>1283</td><td>623</td></tr>
	<tr><th scope=row>PCC_DeJager_eQTL</th><td> 72</td><td>  69</td><td>  15</td><td>3374</td><td>2275</td><td>1747</td><td>2161</td><td>1321</td><td>483</td><td>1120</td><td>587</td></tr>
	<tr><th scope=row>Exc_DeJager_eQTL</th><td>675</td><td> 399</td><td> 409</td><td>  98</td><td>2992</td><td>1769</td><td>2026</td><td> 882</td><td>413</td><td>1203</td><td>665</td></tr>
	<tr><th scope=row>Inh_DeJager_eQTL</th><td>458</td><td> 290</td><td> 305</td><td>  70</td><td>  86</td><td>1644</td><td>1842</td><td> 462</td><td>337</td><td>1150</td><td>636</td></tr>
	<tr><th scope=row>Oli_DeJager_eQTL</th><td>304</td><td> 206</td><td> 230</td><td> 231</td><td> 128</td><td> 172</td><td>1617</td><td> 357</td><td>304</td><td>1076</td><td>536</td></tr>
	<tr><th scope=row>Ast_DeJager_eQTL</th><td>297</td><td> 205</td><td> 226</td><td> 194</td><td> 111</td><td>  97</td><td>  58</td><td> 430</td><td>290</td><td>1200</td><td>555</td></tr>
	<tr><th scope=row>DLPFC_Bennett_pQTL</th><td>182</td><td> 160</td><td> 114</td><td> 271</td><td> 227</td><td> 165</td><td> 167</td><td> 391</td><td>173</td><td> 189</td><td>138</td></tr>
	<tr><th scope=row>monocyte_ROSMAP_eQTL</th><td>146</td><td> 139</td><td> 137</td><td> 159</td><td> 105</td><td>  76</td><td>  81</td><td>  78</td><td> 42</td><td> 174</td><td>272</td></tr>
	<tr><th scope=row>OPC_DeJager_eQTL</th><td>215</td><td> 135</td><td> 152</td><td> 112</td><td>  59</td><td>  43</td><td>  43</td><td> 107</td><td> 47</td><td>  77</td><td>393</td></tr>
	<tr><th scope=row>Mic_DeJager_eQTL</th><td>164</td><td> 142</td><td> 128</td><td> 109</td><td>  63</td><td>  74</td><td>  59</td><td>  89</td><td> 24</td><td>  35</td><td> 85</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="id2">
<h3>heatmap<a class="headerlink" href="#id2" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))),</span><span class="w"> </span>
<span class="w">                           </span><span class="nf">order</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)))]</span>
<span class="n">rev_trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[,</span><span class="nf">ncol</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="c1">#rev_trait_matrix &lt;- t(trait_matrix)</span>
<span class="n">rev_trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_trait_matrix</span><span class="p">)</span>

<span class="c1"># Calculate the fraction matrix</span>
<span class="c1"># Assuming rev_trait_matrix is your input square matrix</span>
<span class="n">context_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">rev_trait_matrix</span><span class="p">)</span>
<span class="n">n</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">context_names</span><span class="p">)</span>

<span class="c1"># Initialize output matrix</span>
<span class="n">proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">context_names</span><span class="w"> </span>

<span class="n">pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">context_names</span><span class="w"> </span>

<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">seq_len</span><span class="p">(</span><span class="n">n</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">seq_len</span><span class="p">(</span><span class="n">n</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="n">j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="c1"># Extract 2x2 submatrix</span>
<span class="w">      </span><span class="n">trait_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)[</span><span class="n">i</span><span class="p">]</span>
<span class="w">      </span><span class="n">trait_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)[</span><span class="n">j</span><span class="p">]</span>
<span class="w">      </span><span class="n">pattern</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_c</span><span class="p">(</span><span class="n">trait_i</span><span class="p">,</span><span class="w"> </span><span class="n">trait_j</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;|&quot;</span><span class="p">)</span>
<span class="w">      </span><span class="n">submat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">)]</span>
<span class="w">      </span><span class="n">ROSMAP_multicontext_res_filtered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="n">pattern</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                                        </span><span class="nf">select</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]])</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">unique</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span>
<span class="w">      </span><span class="n">pairwise_total_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">submat</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="nf">diag</span><span class="p">(</span><span class="n">submat</span><span class="p">))</span>
<span class="w">      </span><span class="n">total_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise_total_sum</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="nf">dim</span><span class="p">(</span><span class="n">ROSMAP_multicontext_res_filtered</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span>
<span class="w">      </span><span class="n">value</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span>

<span class="w">      </span><span class="c1"># Compute and assign proportion</span>
<span class="w">      </span><span class="n">proportion_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">round</span><span class="p">((</span><span class="n">value</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_sum</span><span class="p">)</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
<span class="w">      </span><span class="n">pairwise_proportion_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">round</span><span class="p">((</span><span class="n">value</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">pairwise_total_sum</span><span class="p">)</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>

<span class="c1">#proportion_matrix</span>
<span class="n">rev_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">proportion_matrix</span><span class="p">[,</span><span class="w"> </span><span class="nf">ncol</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="n">rev_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)</span>

<span class="c1"># Convert matrix into long format for ggplot</span>
<span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">reshape2</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Trait1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Trait2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Count&quot;</span><span class="p">)</span>

<span class="c1"># 1. Identify which rows need filling</span>
<span class="n">na_idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">is.na</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Count</span><span class="p">)</span>

<span class="c1"># 2. Pull from the matrix by name for those rows</span>
<span class="c1"># Find numeric row/col indices</span>
<span class="n">r</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">match</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Trait1</span><span class="p">[</span><span class="n">na_idx</span><span class="p">],</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))</span>
<span class="n">c</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">match</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Trait2</span><span class="p">[</span><span class="n">na_idx</span><span class="p">],</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))</span>

<span class="c1"># Do element-wise lookup</span>
<span class="n">df</span><span class="o">$</span><span class="n">Count</span><span class="p">[</span><span class="n">na_idx</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">cbind</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="w"> </span><span class="n">c</span><span class="p">)]</span>

<span class="c1">#proportion_matrix</span>

<span class="n">rev_pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise_proportion_matrix</span><span class="p">[,</span><span class="w"> </span><span class="nf">ncol</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="n">rev_pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_pairwise_proportion_matrix</span><span class="p">)</span>
<span class="w"> </span>
<span class="n">df</span><span class="o">$</span><span class="n">Fraction</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.vector</span><span class="p">(</span><span class="n">reshape2</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">rev_pairwise_proportion_matrix</span><span class="p">)</span><span class="o">$</span><span class="n">value</span><span class="p">)</span>

<span class="n">credible_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">select</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]])</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">unique</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">nrow</span><span class="p">()</span>

<span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">df</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">rowwise</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">Label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Fraction</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">paste0</span><span class="p">(</span><span class="n">Fraction</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n(&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">Count</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;)&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">total_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">Trait1</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">select</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]])</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">unique</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">nrow</span><span class="p">()</span>
<span class="w">      </span>
<span class="w">      </span><span class="nf">round</span><span class="p">((</span><span class="n">Count</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_genes</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">as.character</span><span class="p">()</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">ungroup</span><span class="p">()</span>

<span class="n">df</span><span class="o">$</span><span class="n">Trait1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Trait1</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rev</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)))</span>
<span class="n">df</span><span class="o">$</span><span class="n">Trait2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Trait2</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">))</span>

<span class="c1">#pdf(&quot;ROSMAP_multicontext_effect_sign_heatmap_4063_genes.pdf&quot;, width =10, height = 8)</span>
<span class="c1"># Plot heatmap using ggplot2</span>
<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Trait1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Trait2</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Fraction</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_tile</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_fill_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">na.value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Label</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="c1"># Add text with count and fraction</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Proportion&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Multicontext Posterior Effect Sign Heatmap&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">45</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">),</span>
<span class="w">        </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">),</span>
<span class="w">        </span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">25</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">        </span><span class="n">legend.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">       </span><span class="n">legend.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">))</span>
<span class="c1">#dev.off()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">14</span><span class="p">)</span>
<span class="n">p2</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/3d7ca3de02b8bca8e5a86dc037adc8c99be18038055bfb8f1b407fbf38e50e22.png"><img alt="../../../_images/3d7ca3de02b8bca8e5a86dc037adc8c99be18038055bfb8f1b407fbf38e50e22.png" src="../../../_images/3d7ca3de02b8bca8e5a86dc037adc8c99be18038055bfb8f1b407fbf38e50e22.png" style="width: 1080px; height: 840px;" />
</a>
</div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./main_text/4_multi_context/staging"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#marginal-effect-sign-heatmap">Marginal effect sign heatmap</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#posterior-table">posterior table</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#trait-matrix">trait_matrix</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#heatmap">heatmap</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multicontext-effect-sign-heatmap">multicontext effect sign heatmap</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">trait_matrix</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">heatmap</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>